---
title: "Cell-cell Communication"
author: "Ha Tran"
date: "2024-07-01"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  echo = TRUE,
  message = FALSE,
  warning = FALSE,
  fig.align = "center"
)
```

# Data Setup

```{r load libraries}
# working with data
library(showtext)
library(dplyr)
library(magrittr)
library(readr)
library(tibble)
library(reshape2)
library(tidyverse)

library(pheatmap)
library(ComplexHeatmap)
library(DT)
library(extrafont)
library(readxl)
library(patchwork)
library(edgeR)
library(CellChat)

# Visualisation:
library(VennDiagram)
library(kableExtra)
library(ggplot2)
library(grid)
library(pander)
library(cowplot)
library(viridis)
library(igraph)
library(ggalluvial)
library(RColorBrewer)
library(qs)
# Custom ggplot
library(ComplexHeatmap)
library(ggplotify)
library(ggbiplot)
library(ggrepel)
library(pandoc)
library(knitr)
opts_knit$set(progress = FALSE, verbose = FALSE)
opts_chunk$set(warning=FALSE, message=FALSE, echo=FALSE)
```


```{r importData}
# load DGElist previously created in the set up
dge <- readRDS(here::here("0_data/rds_objects/dge.rds"))
# fc <- readRDS(here::here("0_data/rds_objects/fc.rds"))
# fdr <- readRDS(here::here("0_data/rds_objects/fdr.rds"))
# lm <- readRDS(here::here("0_data/rds_objects/lm.rds"))
lm_all <- readRDS(here::here("0_data/rds_objects/lm_all.rds"))
lm_sig <- readRDS(here::here("0_data/rds_objects/lm_sig.rds"))

# to increase the knitting speed. change to T to save all plots
savePlots <- T
export <- T
```

```{r importFunctions}
# Theme
bossTheme <- readRDS(here::here("0_data/functions/bossTheme.rds"))
bossTheme_bar <- readRDS(here::here("0_data/functions/bossTheme_bar.rds"))
groupColour <- readRDS(here::here("0_data/functions/groupColour.rds"))
groupColour_dark <- readRDS(here::here("0_data/functions/groupColour_dark.rds"))
expressionCol <- readRDS(here::here("0_data/functions/expressionCol.rds"))
expressionCol_dark <- readRDS(here::here("0_data/functions/expressionCol_dark.rds"))
compColour <- readRDS(here::here("0_data/functions/compColour.rds"))

DT <- readRDS(here::here("0_data/functions/DT.rds"))

firstCap <- function(x) {
  substr(x, 1, 1) <- toupper(substr(x, 1, 1))
  x
}

# Plotting
convert_to_superscript <- readRDS(here::here("0_data/functions/convert_to_superscript.rds"))
exponent <- readRDS(here::here("0_data/functions/exponent.rds"))
format_y_axis <- readRDS(here::here("0_data/functions/format_y_axis.rds"))
```

## CellChat analysis {.tabset .tabset-pills}

```{r CellChat, eval=FALSE}
# cpm normalised count data as input for CellChat. Only use DT and DT+Treg group for now
count.data_veh <- cpm(dge) %>% as.data.frame() %>% select_if(grepl("veh",colnames(.)))
count.data_dt <- cpm(dge) %>% as.data.frame() %>% select_if(grepl("DT\\d",colnames(.)))
count.data_dtTreg <- cpm(dge) %>% as.data.frame() %>% select_if(grepl("DT\\+",colnames(.)))

# for cell chat, samples ID will be considered as individual celltype and groups will be considered sample ID
meta_veh <- data.frame(samples = rep("veh",6),labels = colnames(count.data_veh),row.names = colnames(count.data_veh))
meta_dt <- data.frame(samples = rep("DT",5),labels = colnames(count.data_dt),row.names = colnames(count.data_dt))
meta_dtTreg <- data.frame(samples = rep("DT+Treg",5),labels = colnames(count.data_dtTreg),row.names = colnames(count.data_dtTreg))

# create cell Chat object
cellchat_veh <- createCellChat(object = count.data_veh %>% as.matrix(),meta = meta_veh, group.by = "labels")
cellchat_dt <- createCellChat(object = count.data_dt %>% as.matrix(),meta = meta_dt , group.by = "labels")
cellchat_dtTreg <- createCellChat(object = count.data_dtTreg %>% as.matrix(),meta = meta_dtTreg, group.by = "labels")

CellChatDB <- CellChatDB.mouse
# showDatabaseCategory(CellChatDB)

# use all CellChatDB except for "Non-protein Signaling" for cell-cell communication analysis
CellChatDB.use <- subsetDB(CellChatDB, search = "Secreted Signaling", key = "annotation") # use Secreted Signaling

cellchat_veh@DB <- CellChatDB
cellchat_dt@DB <- CellChatDB

# subset the expression data of signaling genes for saving computation cost. subsets the expression data by the signalling genes CellChatDB.use
cellchat_veh <- subsetData(cellchat_veh) # This step is necessary even if using the whole database
cellchat_dt <- subsetData(cellchat_dt) # This step is necessary even if using the whole database

future::plan("multisession", workers = 4) # do parallel
# cellchat_veh <- identifyOverExpressedGenes(cellchat_veh,thresh.p = 0.25)
cellchat_veh <- identifyOverExpressedInteractions(cellchat_veh)


# project gene expression data onto PPI (Optional: when running it, USER should set `raw.use = FALSE` in the function `computeCommunProb()` in order to use the projected data)
# cellchat <- projectData(cellchat, PPI.mouse)
cellchat_veh <- computeCommunProb(cellchat_veh, type = "triMean", raw.use = T)
# cellchat <- filterCommunication(cellchat)
df.net_veh <- subsetCommunication(cellchat_veh)
cellchat_veh <- computeCommunProbPathway(cellchat_veh)
cellchat_veh <- aggregateNet(cellchat_veh)
groupSize <- as.numeric(table(cellchat_veh@idents))

qsave(cellchat_veh, here::here("0_data/rds_objects/cellchat_veh.qs"))
```

# Single cell paper

```{r, eval=FALSE}
scFull <- read_excel(path = here::here("0_data/rawData/Table S1 13578_2022_749_MOESM1_ESM (1).xlsx"), sheet = "Table-S1-d6", skip = 2) %>% dplyr::filter(cell_type %in% c("T","NK","NKp")) %>%  type.convert(.,as.is = F)

sc <- scFull %>% group_by(cell_type) %>% group_split(.) %>% setNames(levels(scFull$cell_type))

# venn.diagram(x= list( "NK" = sc[[1]]$gene_symbol %>% as.character(),
                      # "NKp"= sc[[2]]$gene_symbol %>% as.character(),
                      # "T"  = sc[[3]]$gene_symbol %>% as.character()),disable.logging = T,filename = NULL) %>% grid.draw()

extract_substrings <- function(text) {

  # substrings <- gsub("\\+", " ", text)          # Replace '+' with space
  # substrings <- gsub("-", " ", substrings)            # Replace '-' with space
  substrings <- gsub(",", "", text)
  # Split words if there's a space
  words <- unlist(strsplit(substrings, " "))
  
  return(words)
}

# Separate the ligand and receptor gene symbol.
CellChatDB <- CellChatDB.mouse$interaction
CellChatDB$ligand.symbol_2 <- sapply(CellChatDB$ligand.symbol, extract_substrings)
CellChatDB$receptor.symbol_2 <- sapply(CellChatDB$receptor.symbol, extract_substrings)
CellChatDB <- unnest(CellChatDB, ligand.symbol_2)
CellChatDB <- unnest(CellChatDB, receptor.symbol_2)


t  <- scFull %>% dplyr::filter(gene_symbol %in% CellChatDB$ligand.symbol_2 | gene_symbol %in% CellChatDB$receptor.symbol_2)
t2 <- scFull %>% dplyr::filter(gene_symbol %in% CellChatDB$ligand.symbol_2 & gene_symbol %in% CellChatDB$receptor.symbol_2)
t3 <- scFull %>%
  filter(cell_type == "T" & gene_symbol %in% CellChatDB$ligand.symbol_2) %>% 
  filter(cell_type == "NK" & gene_symbol %in% CellChatDB$receptor.symbol_2)
  
t4 <- scFull %>%
  filter(cell_type == "NK" & gene_symbol %in% CellChatDB$ligand.symbol_2) %>% 
  filter(cell_type == "T" & gene_symbol %in% CellChatDB$receptor.symbol_2)


db_T.NK <- CellChatDB %>% 
  filter(ligand.symbol_2 %in% sc[["T"]]$gene_symbol & receptor.symbol_2 %in% sc[["NK"]]$gene_symbol)

db_T.pNK <- CellChatDB %>% 
  filter(ligand.symbol_2 %in% sc[["T"]]$gene_symbol & receptor.symbol_2 %in% sc[["NKp"]]$gene_symbol)

db_NK.T <- CellChatDB %>% filter(ligand.symbol_2 %in% sc[["NK"]]$gene_symbol & receptor.symbol_2 %in% sc[["T"]]$gene_symbol)

scFull_subsetByDB <- scFull %>% dplyr::filter(gene_symbol %in% CellChatDB$ligand | gene_symbol %in% CellChatDB$receptor)
                                              
sc_subsetByDB <- scFull_subsetByDB %>% group_by(cell_type) %>% group_split(.) %>% setNames(levels(scFull_subsetByDB$cell_type))

# venn.diagram(x= list( "NK" = sc_subsetByDB[[1]]$gene_symbol %>% as.character(),
#                       "NKp"= sc_subsetByDB[[2]]$gene_symbol %>% as.character(),
#                       "T"  = sc_subsetByDB[[3]]$gene_symbol %>% as.character()),disable.logging = T,filename = NULL) %>% grid.draw()

CellChatDB_subsetByscFull <- CellChatDB %>% dplyr::filter(ligand %in% scFull$gene_symbol | receptor %in% scFull$gene_symbol) %>%  dplyr::select(-receptor) %>% dplyr::distinct()




```

```{r}
logCPM_combined <- cpm(dge, prior.count=3, log=TRUE)
rownames(logCPM_combined) <- dge$genes$gene


#merge the log cpm counts with the top 30 common de genes
logCPM_combined <- logCPM_combined[c("H2-T23","Klrd1","Thy1","Itgb2","Itgax","Adgre5"),]
logCPM_combined <- logCPM_combined[,c(6,7,8,9,10,11,1,2,3,4,5,12,13,14,15,16)]

#df for heatmap annotation of sample type
anno_combined <- factor(dge$samples$group, levels = c("veh", "DT", "DT+Treg")) %>% as.data.frame()
anno_combined <- anno_combined[c(6,7,8,9,10,11,1,2,3,4,5,12,13,14,15,16),]%>% as.data.frame()

colnames(anno_combined) <- "Groups"


heat_combined <- ComplexHeatmap::pheatmap(
    mat = logCPM_combined,
    color = colorRampPalette(rev(c("#FB8072","#FDB462","#ffffd5","#8DD3C7","#80B1D3")))(300),
    scale = "row",
    cluster_cols = F,
    cluster_rows = F,
    border_color = "white",
    gaps_col = c(6,11),
    gaps_row = c(2),
    # cutree_cols = 3,
    # cutree_rows = 6,
    # treeheight_row = 40,
    # treeheight_col = 30,
    show_colnames = T,
    clustering_distance_rows = "euclidean",
    # main = paste0("Top ", nrow(logCPM_combined), " significant DEGs"),


    legend = T,

    heatmap_legend_param = list(title = "Expression\nZ-score",
                                direction= "vertical",
                                merge_legend = T,
                                legend_direction = "vertical",
                                legend_width = unit(5, "cm")),

    # annotation = T,
    annotation_legend = T,
    annotation_col = anno_combined,
    annotation_colors = list("Groups" = groupColour),
    annotation_names_col = T,
    # annotation_legend_param = list(direction = "horizontal"),

    fontfamily = "Arial Narrow",
    fontsize = 12,
    fontsize_col = 12,
    fontsize_number = 12,
    fontsize_row = 12,
    labels_row = as.expression(lapply(rownames(logCPM_combined), function(a) bquote(italic(.(a)))))
  )

draw(heat_combined, merge_legend = T, heatmap_legend_side = "right",
    annotation_legend_side = "right")

ggsave(filename = "LRheat.svg", plot = heat_combined, path = here::here("2_plots/"),width = 25.5,height = 20,units = "cm")

```

```{r}
lm_all <- readRDS(here::here("0_data/rds_objects/lm_all.rds"))
lm_all_LR <- lapply(lm_all, function(x) {
  x %>% dplyr::filter(symbol %in% c("H2-T23","Thy1","Klrd1","Itgb2","Itgax","Adgre5")) %>% dplyr::select(c("gene","logFC")) 
})

LR_mat <- lm_all_LR[[1]] %>% 
  left_join(lm_all_LR[[3]], by = "gene") %>% column_to_rownames("gene") %>% magrittr::set_colnames(c("DT vs veh", "DT+Treg vs DT"))


LR_heat <- ComplexHeatmap::pheatmap(
    mat = LR_mat,
    color = colorRampPalette(rev(c("#FB8072","#FDB462","#ffffd5","#8DD3C7","#80B1D3")))(300),
    scale = "none",
    cluster_cols = F,
    border_color = "white",
    # gaps_col = c(6,11),
    # cutree_cols = 3,
    # cutree_rows = 6,
    # treeheight_row = 40,
    # treeheight_col = 30,
    show_colnames = T,
    clustering_distance_rows = "euclidean",
    # main = paste0("Top ", nrow(logCPM_combined), " significant DEGs"),


    legend = T,

    heatmap_legend_param = list(title = "Expression\nZ-score",
                                direction= "vertical",
                                merge_legend = T,
                                legend_direction = "vertical",
                                legend_width = unit(5, "cm")),

    # annotation = T,
    

    fontfamily = "Arial Narrow",
    fontsize = 12,
    fontsize_col = 12,
    fontsize_number = 12,
    fontsize_row = 12,
    labels_row = as.expression(lapply(rownames(logCPM_combined), function(a) bquote(italic(.(a)))))
  )

draw(LR_heat, merge_legend = T, heatmap_legend_side = "right", 
    annotation_legend_side = "right")
LR_mat <- heatMatrix_anno %>% 
  dplyr::mutate_if(is.numeric, funs(as.character(signif(.,3)))) %>% 
  dplyr::mutate(across(c(`DT vs veh`,`DT+Treg vs DT`), ~ case_when(. < 0.001 ~ "****",
                                                                   . < 0.01 ~ "***",
                                                                   . < 0.05 ~ "**",
                                                                   . < 0.1 ~ "*",
                                                                   TRUE ~ as.character(.))))
```



```{r}
cellchat <- qread(here::here("0_data/rds_objects/cellchat.qs"))
par(mfrow = c(1,2), xpd=TRUE)
netVisual_circle(cellchat_veh@net$count, vertex.weight = groupSize, weight.scale = T, label.edge= F, title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight, vertex.weight = groupSize, weight.scale = T, label.edge= F, title.name = "Interaction weights/strength")

mat <- cellchat@net$weight
par(mfrow = c(2,5), xpd=TRUE)
for (i in 1:nrow(mat)) {
  mat2 <- matrix(0, nrow = nrow(mat), ncol = ncol(mat), dimnames = dimnames(mat))
  mat2[i, ] <- mat[i, ]
  netVisual_circle(mat2, vertex.weight = groupSize, weight.scale = T, edge.weight.max = max(mat), title.name = rownames(mat)[i])
}


pathways.show <- c("TGFb") 
# Hierarchy plot
# Here we define `vertex.receive` so that the left portion of the hierarchy plot shows signaling to fibroblast and the right portion shows signaling to immune cells 
vertex.receiver = seq(1,4) # a numeric vector. 
netVisual_aggregate(cellchat, signaling = pathways.show,  vertex.receiver = vertex.receiver)
# Circle plot
par(mfrow=c(1,1))
netVisual_aggregate(cellchat, signaling = pathways.show, layout = "circle")

par(mfrow=c(1,1))
netVisual_aggregate(cellchat, signaling = pathways.show, layout = "chord")

par(mfrow=c(1,1))
# netVisual_heatmap(cellchat,, color.heatmap = "Reds")
#> Do heatmap based on a single object

group.cellType <- c(rep("DT",5), rep("DT+Treg",5)) # grouping cell clusters into fibroblast, DC and TC cells
names(group.cellType) <- levels(cellchat@idents)
netVisual_chord_cell(cellchat, signaling = pathways.show, group = group.cellType, title.name = paste0(pathways.show, " signaling network"))
#> Plot the aggregated cell-cell communication network at the signaling pathway level


netAnalysis_contribution(cellchat, signaling = pathways.show)

# (1) show all the significant interactions (L-R pairs) from some cell groups (defined by 'sources.use') to other cell groups (defined by 'targets.use')
netVisual_bubble(cellchat, sources.use = 4, targets.use = c(5:11), remove.isolate = FALSE)
#> Comparing communications on a single object


cellchat <- netAnalysis_computeCentrality(cellchat, slot.name = "netP") # the slot 'netP' means the inferred intercellular communication network of signaling pathways
# Visualize the computed centrality scores using heatmap, allowing ready identification of major signaling roles of cell groups
netAnalysis_signalingRole_network(cellchat, signaling = pathways.show, width = 8, height = 2.5, font.size = 10)

```





# Trophoblast SC paper

```{r trophoblastScPaper, eval=F}
JiangSC <- read_csv(here::here("0_data/rawData/Jiang_TableS2_41421_2022_513_MOESM3_ESM.csv"))

lm_sig_JiangSC_subset <- lapply(seq_along(lm_sig), function(x){
  lm_sig[[x]] %>% dplyr::filter(gene %in% JiangSC$gene) %>% merge(.,JiangSC[,c("gene", "cluster")], by= "gene") %>% 
    dplyr::mutate(comp = names(lm_sig)[x],
                  clustAnno = case_when(cluster == 1 ~ "TSCs & ExE",
                                        cluster == 2 | cluster == 3 ~ "LaTP",
                                        cluster == 4 | cluster == 5 ~ "SynTI precusor",
                                        cluster == 6 | cluster == 7 | cluster == 8 ~ "S-TGC precusor",
                                        cluster == 9 ~ "S-TGC",
                                        cluster == 10 ~ "1' P-TGC",
                                        cluster == 11 ~ "2' P-TGC",
                                        cluster == 12 ~ "2' P-TGC precusor",
                                        cluster == 13 ~ "EPC cell",
                                        cluster == 14 | cluster == 15 ~ "SpT",
                                        cluster == 16 ~ "Gly-T",
                                        cluster == 17 ~ "Spa-TGC",
                                        cluster == 18 ~ "EPC migratory cell",
                                        cluster == 19 ~ "SynTII precusor"))
}) %>% setNames(.,names(lm_sig))

# lapply(lm_sig_JiangSC_subset, function(x) x %>% dplyr::select(c("gene", "logFC", "AveExpr", "P.Value", "adj.P.Val", "description", "entrezid","expression", "cluster"))) %>% writexl::write_xlsx(x = ., here::here("3_output/sig_de_JiangSC_subset.xlsx"))

subSet_lm_sig <- rbind(lm_sig_JiangSC_subset[[1]],lm_sig_JiangSC_subset[[3]])

```


```{r option1}
logCPM_combined <- cpm(dge, prior.count=3, log=TRUE)
rownames(logCPM_combined) <- dge$genes$gene


#merge the log cpm counts with the top 30 common de genes
logCPM_combined <- logCPM_combined[subSet_lm_sig$gene %>% as.factor() %>% levels(),]
logCPM_combined <- logCPM_combined[,c(6,7,8,9,10,11,1,2,3,4,5,12,13,14,15,16)]

#df for heatmap annotation of sample type
anno_combined <- factor(dge$samples$group, levels = c("veh", "DT", "DT+Treg")) %>% as.data.frame()
anno_combined <- anno_combined[c(6,7,8,9,10,11,1,2,3,4,5,12,13,14,15,16),]%>% as.data.frame()

colnames(anno_combined) <- "Groups"


# heat_combined <- 
  ComplexHeatmap::pheatmap(
    mat = logCPM_combined,
    color = colorRampPalette(rev(c("#FB8072","#FDB462","#ffffd5","#8DD3C7","#80B1D3")))(300),
    scale = "row",
    cluster_cols = F,
    cluster_rows = T,
    border_color = "white",
    gaps_col = c(6,11),
    # gaps_row = c(2),
    # cutree_cols = 3,
    # cutree_rows = 6,
    # treeheight_row = 40,
    # treeheight_col = 30,
    show_colnames = T,
    clustering_distance_rows = "euclidean",
    # main = paste0("Top ", nrow(logCPM_combined), " significant DEGs"),


    legend = T,

    heatmap_legend_param = list(title = "Expression\nZ-score",
                                direction= "vertical",
                                merge_legend = T,
                                legend_direction = "vertical",
                                legend_width = unit(5, "cm")),

    # annotation = T,
    annotation_legend = T,
    annotation_col = anno_combined,
    annotation_colors = list("Groups" = groupColour),
    annotation_names_col = T,
    # annotation_legend_param = list(direction = "horizontal"),

    fontfamily = "Arial Narrow",
    fontsize = 12,
    fontsize_col = 12,
    fontsize_number = 12,
    fontsize_row = 12,
    labels_row = as.expression(lapply(rownames(logCPM_combined), function(a) bquote(italic(.(a)))))
  ) %>% as.ggplot()




```

```{r option2}
# Option 2.1
mat <- subSet_lm_sig %>% dplyr::select(gene, logFC, comp, clustAnno) %>% distinct() %>%  pivot_wider(names_from = comp, values_from = logFC) %>% 
  group_by(gene) %>%
  arrange(clustAnno) %>% 
  mutate(clustAnno_id = paste0("clustAnno", row_number())) %>%
  ungroup() %>% 
  pivot_wider(names_from = clustAnno_id, values_from = clustAnno, names_prefix = "")

mat_label_anno <- merge(mat,lm_all$`DT+Treg vs DT`[lm_all$`DT+Treg vs DT`$gene %in% mat$gene,c("gene","adj.P.Val")], by = "gene") %>% mutate(`DT+Treg vs DT` = adj.P.Val) %>% dplyr::select(-adj.P.Val) %>%
  merge(.,lm_all$`DT vs veh`[lm_all$`DT vs veh`$gene %in% mat$gene,c("gene","adj.P.Val")], by = "gene") %>% mutate(`DT vs veh` = adj.P.Val) %>% dplyr::select(-adj.P.Val) %>%  group_by(gene) %>%
  arrange(clustAnno1) %>% 
  dplyr::mutate_if(is.numeric, funs(as.character(signif(.,3)))) %>% 
  dplyr::mutate(across(c(`DT vs veh`,`DT+Treg vs DT`), ~ case_when(. < 0.001 ~ "****",
                                                                   . < 0.01 ~ "***",
                                                                   . < 0.05 ~ "**",
                                                                   . < 0.1 ~ "*",
                                                                   TRUE ~ as.character(""))))

mat <- merge(mat,lm_all$`DT+Treg vs DT`[lm_all$`DT+Treg vs DT`$gene %in% mat$gene,c("gene","logFC")], by = "gene") %>% mutate(`DT+Treg vs DT` = logFC) %>% dplyr::select(-logFC) %>%
  merge(.,lm_all$`DT vs veh`[lm_all$`DT vs veh`$gene %in% mat$gene,c("gene","logFC")], by = "gene") %>% mutate(`DT vs veh` = logFC) %>% dplyr::select(-logFC) %>%  group_by(gene) %>%
  arrange(clustAnno1)

mat[,c(1,4:ncol(mat))][is.na(mat[, c(1,4:ncol(mat))])] <- "empty"

rowAnno_colours <- c("white",colorRampPalette(brewer.pal(9, "Set1"))(14))
names(rowAnno_colours) <- c("empty",levels(subSet_lm_sig$clustAnno %>% as.factor()))

m <- function(range){
  hm <- pheatmap(
      # MAIN
      mat = mat[range,1:3] %>% column_to_rownames("gene") %>% as.matrix(),
      # display_numbers = mat1_anno,
      color = colorRampPalette(rev(c("#FB8072","#FDB462","grey95","#8DD3C7","#80B1D3")))(length(seq(-3,3, length.out=(4 + 1)))),
      cellwidth = 30,
      scale = "none",
      
      # Col
      cluster_cols = F,
      border_color = "white",
      angle_col = "45",
      # gaps_row = c(23,30,38),
      
      # Row
      cluster_rows = F,
      display_numbers = mat_label_anno[range,1:3] %>% column_to_rownames("gene") %>% as.matrix(),
      
      # Labs
      show_colnames = T,
      show_rownames = T,
      legend = F,
      breaks = seq(-3,3, length.out=(4 + 1)),
      heatmap_legend_param = list(title = "logFC", legend_width = unit(7, "cm")),
      
      # Annotation
      annotation_legend = F,
      # legend_labels = T,
      annotation_row = mat[range,c(1,4:ncol(mat))] %>% column_to_rownames("gene") %>% as.data.frame(),
      annotation_colors = list("clustAnno1" = rowAnno_colours,
                               "clustAnno2" = rowAnno_colours,
                               "clustAnno3" = rowAnno_colours,
                               "clustAnno4" = rowAnno_colours,
                               "clustAnno5" = rowAnno_colours,
                               "clustAnno6" = rowAnno_colours,
                               "clustAnno7" = rowAnno_colours,
                               "clustAnno8" = rowAnno_colours),
      
      annotation_names_row =F,
      
      # Fonts
      fontfamily = "Arial Narrow",
      fontsize = 12,
      fontsize_col = 12,
      fontsize_number =12,
      fontsize_row = 10
      
    ) %>% as.ggplot()
  return(hm)
}

t <- m(1:43) + plot_spacer() + m(44:86) + plot_spacer() + m(87:129) +
  plot_layout(widths = c(3,-1.5,3,-1.5,3))

ggsave(filename = "trophoblast_hmap_5.svg", plot = t, path = here::here("2_plots/"),
           width = 21, height = 27, units = "cm")

# Option 2.2
mat <- subSet_lm_sig %>% select(gene, logFC, comp, clustAnno) %>% distinct() %>%  pivot_wider(names_from = comp, values_from = logFC) %>% na.omit() %>%
  group_by(gene) %>%
  arrange(clustAnno) %>%
  mutate(clustAnno_id = paste0("clustAnno", row_number())) %>%
  ungroup() %>% 
  pivot_wider(names_from = clustAnno_id, values_from = clustAnno, names_prefix = "")

mat[,c(1,4:ncol(mat))][is.na(mat[, c(1,4:ncol(mat))])] <- "empty"

m <- function(range){
  hm <- pheatmap(
      # MAIN
      mat = mat[range,1:3] %>% column_to_rownames("gene") %>% as.matrix(),
      # display_numbers = mat1_anno,
      color = colorRampPalette(rev(c("#FB8072","#FDB462","grey95","#8DD3C7","#80B1D3")))(100),
      cellwidth = 70,
      scale = "none",
      
      # Col
      cluster_cols = F,
      border_color = "white",
      angle_col = "0",
      # gaps_row = c(23,30,38),
      
      # Row
      cluster_rows = F,
      
      # Labs
      show_colnames = T,
      show_rownames = T,
      legend = T,
      breaks = seq(-2,2, length.out=(2 + 1)),
      heatmap_legend_param = list(title = "logFC", legend_width = unit(7, "cm")),
      
      # Annotation
      annotation_legend = T,
      # legend_labels = T,
      annotation_row = mat[range,c(1,4:ncol(mat))] %>% column_to_rownames("gene") %>% as.data.frame(),
      annotation_colors = list("clustAnno1" = rowAnno_colours,
                               "clustAnno2" = rowAnno_colours,
                               "clustAnno3" = rowAnno_colours,
                               "clustAnno4" = rowAnno_colours,
                               "clustAnno5" = rowAnno_colours),
      
      annotation_names_row =F,
      
      # Fonts
      fontfamily = "Arial Narrow",
      fontsize = 12,
      fontsize_col = 12,
      fontsize_number = 8,
      fontsize_row = 10
      
    ) %>% as.ggplot()
  return(hm)
}


t <- m(1:nrow(mat))
# + plot_spacer() + m(44:86) + plot_spacer() + m(87:129) +
#   plot_layout(widths = c(3,-1.5,3,-1.5,3))


# Option 2.3
logCPM_combined <- cpm(dge, prior.count=3, log=TRUE) %>% as.data.frame() %>%  rownames_to_column("gene")
logCPM_combined <- logCPM_combined[logCPM_combined$gene %in% mat$gene,]
logCPM_combined <- logCPM_combined %>% left_join(.,mat[,c(1,4:ncol(mat))] %>% as.data.frame(), by = "gene") %>% column_to_rownames("gene")
logCPM_combined <- logCPM_combined[,c(6,7,8,9,10,11,1,2,3,4,5,12:ncol(logCPM_combined))]

colAnno_combined <- factor(dge$samples$group, levels = c("veh", "DT", "DT+Treg")) %>% as.data.frame()
colAnno_combined <- colAnno_combined[c(6,7,8,9,10,11,1,2,3,4,5,12,13,14,15,16),]%>% as.data.frame()
colnames(colAnno_combined) <- "Groups"



m2 <- function(df){
  hm <- pheatmap(
      # MAIN
      mat = df[,1:16] %>% as.matrix(),
      # display_numbers = mat1_anno,
      color = colorRampPalette(rev(c("#FB8072","#FDB462","grey95","#8DD3C7","#80B1D3")))(100),
      cellwidth = 20,
      scale = "row",
      
      # Col
      cluster_cols = F,
      border_color = "white",
      angle_col = "90",
      gaps_col = c(6,11),
      
      # Row
      cluster_rows = T,
      # gaps_row = c(23,30,38),
      clustering_distance_rows = "euclidean",


      # Labs
      show_colnames = T,
      show_rownames = T,
      legend = T,
      # breaks = seq(-2,2, length.out=(2 + 1)),
      heatmap_legend_param = list(title = "Expression\nZ-score", legend_width = unit(7, "cm")),
      
      # Annotation
      annotation_legend = T,
      # legend_labels = T,
      annotation_row = df[,c(17:ncol(df))] %>%  as.data.frame(),
      annotation_col = colAnno_combined,
      annotation_colors = list("clustAnno1" = rowAnno_colours,
                               "clustAnno2" = rowAnno_colours,
                               "clustAnno3" = rowAnno_colours,
                               "clustAnno4" = rowAnno_colours,
                               "clustAnno5" = rowAnno_colours,
                               "Groups" = groupColour),
      
      annotation_names_col = T,
      annotation_names_row =F,
      
      # Fonts
      fontfamily = "Arial Narrow",
      fontsize = 12,
      fontsize_col = 12,
      fontsize_number = 8,
      fontsize_row = 10
      
    ) %>% as.ggplot()
  return(hm)
} 

option2.3 <- m2(logCPM_combined)

ggsave(filename = "trophoblast_hmap_4.svg", plot = option2.3, path = here::here("2_plots/"),
           width = 21, height = 10, units = "cm")



ggsave(filename = "trophoblast_hmap_3.svg", plot = t, path = here::here("2_plots/"),
           width = 21, height = 10, units = "cm")
ggsave(filename = "trophoblast_hmap_2.svg", plot = t2, path = here::here("2_plots/"),
           width = 21, height = 27, units = "cm")
ggsave(t,)

```

# MGI data

```{r}
mgi <- read_tsv(here::here("0_data/rawData/MGI_tissues.tsv"),col_names = F) %>%
  magrittr::set_colnames(c("gene","description","MGI_ID","species","tissue")) %>% 
  pivot_wider(names_from = "tissue", values_from = "gene") %>% 
  dplyr::filter(is.na(uterus)) %>% dplyr::select(-c(uterus, species))


mgi_mat <- rbind(lm_sig[[1]] %>% filter(gene %in% mgi$placenta) %>% mutate(comp = "DT vs veh"),
                    lm_sig[[3]] %>% filter(gene %in% mgi$placenta) %>% mutate(comp = "DT+Treg vs DT")) %>% 
  dplyr::select(gene, logFC, comp) %>% distinct() %>%  pivot_wider(names_from = comp, values_from = logFC) %>% left_join(., mat[,c(1,4:ncol(mat))], by = "gene") %>% arrange(gene)

mgi_mat <- mgi_mat[,colSums(is.na(mgi_mat)) < nrow(mgi_mat)]

mgi_mat_anno <- merge(mgi_mat,lm_all$`DT+Treg vs DT`[lm_all$`DT+Treg vs DT`$gene %in% mgi_mat$gene,c("gene","adj.P.Val")], by = "gene") %>% mutate(`DT+Treg vs DT` = adj.P.Val) %>% dplyr::select(-adj.P.Val) %>%
  merge(.,lm_all$`DT vs veh`[lm_all$`DT vs veh`$gene %in% mgi_mat$gene,c("gene","adj.P.Val")], by = "gene") %>% mutate(`DT vs veh` = adj.P.Val) %>% dplyr::select(-adj.P.Val) %>% 
  dplyr::mutate_if(is.numeric, funs(as.character(signif(.,3)))) %>% 
  dplyr::mutate(across(c(`DT vs veh`,`DT+Treg vs DT`), ~ case_when(. < 0.001 ~ "****",
                                                                   . < 0.01 ~ "***",
                                                                   . < 0.05 ~ "**",
                                                                   . < 0.1 ~ "*",
                                                                   TRUE ~ as.character(""))))

mgi_mat <- merge(mgi_mat,lm_all$`DT+Treg vs DT`[lm_all$`DT+Treg vs DT`$gene %in% mgi_mat$gene,c("gene","logFC")], by = "gene") %>% mutate(`DT+Treg vs DT` = logFC) %>% dplyr::select(-logFC) %>%
  merge(.,lm_all$`DT vs veh`[lm_all$`DT vs veh`$gene %in% mgi_mat$gene,c("gene","logFC")], by = "gene") %>% mutate(`DT vs veh` = logFC) %>% dplyr::select(-logFC)

mgi_mat[,c(1,4:ncol(mgi_mat))][is.na(mgi_mat[, c(1,4:ncol(mgi_mat))])] <- "empty"

rowAnno_colours <- c("white",colorRampPalette(brewer.pal(9, "Set1"))(14))
names(rowAnno_colours) <- c("empty",levels(subSet_lm_sig$clustAnno %>% as.factor()))

m3 <- function(range){
  hm <- pheatmap(
      # MAIN
      mat = mgi_mat[range,1:3] %>% remove_rownames() %>% column_to_rownames("gene") %>% as.matrix(),
      # display_numbers = mat1_anno,
      color = colorRampPalette(rev(c("#FB8072","#FDB462","grey95","#8DD3C7","#80B1D3")))(length(seq(-3,3, length.out=(4 + 1)))),
      cellwidth = 70,
      scale = "none",
      
      # Col
      cluster_cols = F,
      border_color = "white",
      angle_col = "0",
      # gaps_row = c(23,30,38),
      
      # Row
      cluster_rows = F,
      display_numbers = mgi_mat_anno[range,1:3] %>% remove_rownames() %>%  column_to_rownames("gene") %>% as.matrix(),
      
      # Labs
      show_colnames = T,
      show_rownames = T,
      legend = T,
      breaks = seq(-4,4, length.out=(4 + 1)),
      heatmap_legend_param = list(title = "logFC", legend_width = unit(7, "cm")),
      
      # Annotation
      annotation_legend = T,
      # legend_labels = T,
      annotation_row = mgi_mat[range,c(1,4:ncol(mgi_mat))] %>%  remove_rownames() %>% column_to_rownames("gene") %>% as.data.frame(),
      annotation_colors = list("clustAnno1" = rowAnno_colours,
                               "clustAnno2" = rowAnno_colours,
                               "clustAnno3" = rowAnno_colours,
                               "clustAnno4" = rowAnno_colours,
                               "clustAnno5" = rowAnno_colours),
      
      annotation_names_row =F,
      
      # Fonts
      fontfamily = "Arial Narrow",
      fontsize = 12,
      fontsize_col = 12,
      fontsize_number =12,
      fontsize_row = 10
      
    ) %>% as.ggplot()
  return(hm)
}


mgi_hm <- m3(1:19) + plot_spacer() +  m3(20:38) +
  plot_layout(widths = c(3,-1,3))

ggsave(filename = "mgi_hmap_1.svg", plot = mgi_hm, path = here::here("2_plots/"),
           width = 21, height = 10, units = "cm")
```

