---
title: "DGE Analysis"
author: "Ha Tran"
date: "08-01-2024"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  echo = TRUE,
  message = FALSE,
  warning = FALSE,
  fig.align = "center",
  fig.width = 10
)
```

# Data Setup

```{r load libraries}
# working with data
library(readxl)
library(dplyr)
library(magrittr)
library(readr)
library(tibble)
library(reshape2)
library(tidyverse)
library(ComplexHeatmap)
library(scales)
library(plyr)

# Visualisation:
library(kableExtra)
library(ggplot2)
library(grid)
library(pander)
library(cowplot)
library(pheatmap)
library(VennDiagram)
library(DT)
library(patchwork)
library(kableExtra)
library(extrafont)
loadfonts(device = "all")

# Custom ggplot
library(ggplotify)
library(ggpubr)
library(ggrepel)
library(viridis)

# Bioconductor packages:
library(edgeR)
library(limma)
library(Glimma)

library(pandoc)
library(knitr)
opts_knit$set(progress = FALSE, verbose = FALSE)
opts_chunk$set(warning=FALSE, message=FALSE, echo=FALSE)
```

### Import DGElist Data

DGElist object containing the raw feature count, sample metadata, and gene metadata, created in the Set Up stage.

```{r importData}
# load DGElist previously created in the set up
dge <- readRDS(here::here("0_data/rds_objects/dge.rds"))

# to increase the knitting speed. change to T to save all plots
savePlots <- T
export <- T
```

```{r importFunctions}
# Theme
bossTheme <- readRDS(here::here("0_data/functions/bossTheme.rds"))
bossTheme_bar <- readRDS(here::here("0_data/functions/bossTheme.rds"))
groupColour <- readRDS(here::here("0_data/functions/groupColour.rds"))
groupColour_dark <- readRDS(here::here("0_data/functions/groupColour_dark.rds"))
expressionCol <- readRDS(here::here("0_data/functions/expressionCol.rds"))
expressionCol_dark <- readRDS(here::here("0_data/functions/expressionCol_dark.rds"))


DT <- readRDS(here::here("0_data/functions/DT.rds"))

# Plotting
convert_to_superscript <- readRDS(here::here("0_data/functions/convert_to_superscript.rds"))
exponent <- readRDS(here::here("0_data/functions/exponent.rds"))
format_y_axis <- readRDS(here::here("0_data/functions/format_y_axis.rds"))
```

## Initial Parameterisation {.tabset .tabset-pills}

The varying methods used to identify differential expression all rely on similar initial parameters. These include:

1.  The Design Matrix,

2.  Estimation of Dispersion, and

3.  Contrast Matrix

### Design Matrix

The experimental design can be parameterised in a one-way layout where one coefficient is assigned to each group. The design matrix formulated below contains the predictors of each sample

```{r design}
# null design with unit vector for generation of voomWithQualityWeights downstream
null_design <- matrix(1, ncol = 1, nrow = ncol(dge))

# setup full design matrix with sample_group
full_design <- model.matrix(~ 0 + group,data = dge$samples)

# remove "sample_group" from each column names
colnames(full_design) <- gsub("group","",colnames(full_design))

full_design %>% as.data.frame() %>% DT(., "Table: Design matrix")

```

### Contrast Matrix

The contrast matrix is required to provide a coefficient to each comparison and later used to test for significant differential expression with each comparison group

```{r constrastMatrix}
contrast <- limma::makeContrasts(
  DTvPBS   = DT - PBS,
  TregvPBS = Treg - PBS,
  TregvDT  = Treg - DT,
  levels = full_design)

colnames(contrast) <- c("DT vs PBS", "Treg vs PBS", "Treg vs DT")

contrast %>% DT(., "Table: Contrast matrix")
```

# Limma-Voom

## Apply voom transformation {.tabset .tabset-pills}

Voom is used to estimate the mean-variance relationship of the data, which is then used to calculate and assign a precision weight for each of the observation (gene). This observational level weights are then used in a linear modelling process to adjust for heteroscedasticity. Log count (logCPM) data typically show a decreasing mean-variance trend with increasing count size (expression).

However, for some dataset with potential sample outliers, `voomWithQualityWeights` can be used to calculate sample-specific quality weights. The application of observational and sample-specific weights can objectively and systematically correct for outliers and better than manually removing samples in cases where there are no clear-cut reasons for replicate variations. Thus, linear model will be applied to the voom transformation with observational and sample-specific weights.

### Observational level weights

```{r voom, fig.cap = "Voom transformation with observational weights"}
# voom transformation without sample weights
voom <- limma::voom(counts = dge, design = full_design, plot = TRUE,)
```

### Observational & group level weights

```{r voom_groupWeights, fig.cap = "Voom transformation with observational and group-specific weights"}
# voom transformation with sample weights using full_design matrix for group-specific weights
voom1 <- limma::voomWithQualityWeights(counts = dge, design = full_design, plot = TRUE)
```

### Observational & sample level weights

```{r voom_sampleWeights, fig.cap = "Voom transformation with observational and sample-specific weights"}
# voom transformation with sample weights using null design matrix
voom2 <- limma::voomWithQualityWeights(counts = dge,design = null_design, plot = TRUE)
```

### PCA
```{r voomPCA, fig.width=11,fig.cap="PCA plots after voom quality weight transformations. Group-specific weights(left) and sample-specific weights (right)"}
# Function for performing pca and generating plots
voom_pca <- function(voom_trans, titleIndex){
  title <- c("Group-level weights", "Sample-level weights")
  pca <- voom_trans$E %>% 
    t() %>% 
    prcomp()
  
  voom_trans$targets <- voom_trans$targets %>% as.data.frame %>% rownames_to_column("Sample_Name")
  
  pca$x %>%
    as.data.frame() %>%
    rownames_to_column("Sample_Name") %>%
    as_tibble() %>%
    dplyr::select(Sample_Name, PC1, PC2) %>%
    left_join(voom_trans$targets, by = "Sample_Name") %>%
    mutate(sample.weights = round(sample.weights, 3)) %>%
    ggplot(aes(PC1, PC2, colour = group, size = sample.weights, label = Sample_Name)) +
    geom_label_repel(box.padding = grid::unit(0.5,"lines"), size = 3, label.size = 0.15, show.legend = F) +
    geom_point(alpha = 0.5) +
    labs(x = paste0("PC1 (", percent(summary(pca)$importance["Proportion of Variance","PC1"]),")"),
         y = paste0("PC2 (", percent(summary(pca)$importance["Proportion of Variance","PC2"]),")"),
         colour = "Groups",
         size = "Weights",
         shape = "Replicate",
         title = title[titleIndex]) +
    bossTheme(14,legend = "bottom") +
    guides(colour = guide_legend(override.aes = list(size = 3)))
}

# iterate with lapply for voom 1 and 2
voom_pca_plots <- lapply(1:2, function(y){
  voom_pca(voom_trans = get(paste0('voom',y)), titleIndex = y)
})

# plot pca of voom1 and voom2 side by side
plot_grid(
  plot_grid(
    voom_pca_plots[[1]] + theme(legend.position = "none"),
    voom_pca_plots[[2]] + theme(legend.position = "none"),
    nrow = 1
  ),
  get_legend(voom_pca_plots[[1]] + theme(legend.position = "bottom")),
  nrow = 2,
  rel_heights = c(10,1)
)
```

## Apply linear model {.tabset .tabset-pills}

When the list of DE genes is large, we can apply a fold change cut-off through application of `TREAT` to prioritise the genes with greater fold changes and potentially more biologically relevant. Ideally, we are aiming for \~300 genes $\pm$ 100 genes. Functional enrichment analysis with this number of genes should generate meaningful results.

Importantly, the FC threshold used in `TREAT` should be chosen as a small value below which results should be ignored, instead of a target fold-change. In general, a modest fold-change of 1.1 - 1.5 is recommended. However, it is more important to select a fold-change cut-off that generates a sufficiently small list of DE genes.

A quick aside on the definition and interpretation of fold change and `log2FC`. A fold-change (FC) refers to the **ratio** of two values.

-   If there is a **two** fold increase (`FC = 2`, `log2FC = 1`) between `A vs B`, then `A` is twice as big as `B` (or `A` is `200%` of `B`)
-   If there is a **two** fold decrease (`FC = 0.5`, `log2FC = -1`) between `A vs B`, then `A` is halfas big as `B` (or `A` is `50%` of `B`)

```{r fcAndLfc}
# specifying FC of interest
options(digits = 6)
fc <- c("none",1.1, 1.2, 1.3)
```

### FC=`r fc[1]`

```{r dtTable, results='asis'}
# function for applying linear model, generate decideTest table, and extract topTable
limmaFit <- function(x, fc, adjMethod, p.val, tableNum){
  lm <- limma::lmFit(object = x, design = full_design) %>%
    contrasts.fit(contrasts = contrast)
  
  if (fc == "none") {
    lm <- lm %>% limma::eBayes()
  } else {
    lm <- lm %>% limma::treat(fc = as.numeric(fc))
  }
  
  lm_dt <- decideTests(object = lm, adjust.method = adjMethod, p.value = p.val)
  print(knitr::kable(summary(lm_dt)
                      , caption = paste0("TABLE ",tableNum, ": Number of significant DE genes with '", adjMethod, "' adjusment method, and at a p-value/adj.p-value of ", p.val)) %>% 
    kable_styling(bootstrap_options = c("striped", "hover")))
}

limmaFit(x = voom1, fc[1], adjMethod = "none", p.val = 0.01, 1)

limmaFit(x = voom1, fc[1], adjMethod = "fdr", p.val = 0.1, 2)

limmaFit(x = voom1, fc[1], adjMethod = "fdr", p.val = 0.05, 3)

```

### FC=`r fc[2]`

```{r, results='asis'}
limmaFit(x = voom1, fc[2], adjMethod = "none", p.val = 0.01,4)

limmaFit(x = voom1, fc[2], adjMethod = "fdr", p.val = 0.1, 5)

limmaFit(x = voom1, fc[2], adjMethod = "fdr", p.val = 0.05, 6)
```

### FC=`r fc[3]`

```{r, results='asis'}
limmaFit(x = voom1, fc[3], adjMethod = "none", p.val = 0.01,7)

limmaFit(x = voom1, fc[3], adjMethod = "fdr", p.val = 0.1, 8)

limmaFit(x = voom1, fc[3], adjMethod = "fdr", p.val = 0.05, 9)
```

### FC=`r fc[4]`

```{r, results='asis'}
limmaFit(x = voom1, fc[4], adjMethod = "none", p.val = 0.01,10)

limmaFit(x = voom1, fc[4], adjMethod = "fdr", p.val = 0.1, 11)

limmaFit(x = voom1, fc[4], adjMethod = "fdr", p.val = 0.05, 12)
```

# Differential Gene Expression analysis {.tabset .tabset-pills}

Due to the large variance in the PBS group, the transformation with `observational and group-level weights` were used. **Without** FC cut-off (using TREAT) and an `FDR < 0.05`, the `DT vs Treg` comparison had 45 significant DE genes **(TABLE 3)**. This may **not** be enough DE genes to perform meaningful functional enrichment analysis downstream. Therefore, The FDR threshold is increased to `0.1`, in another word, we allow for 10% type I error, i.e. 1 in 10 genes may be a false positive for differential expression. Using a fold-change cut-off through application of `TREAT` gives additional stringency at the costs of reducing the list of DE genes.

-   **P-value histogram:** illustrates the distribution of p-values. As the stringency increases (increasing FC threshold), the distribution shifts towards `1`, thus insignificant.

-   **MA plot:** helps visualise and identify genes with significant changes in expression. Points deviating from the central axis often indicate differentially expressed genes, allowing assessment of the magnitude and consistency of expression changes across conditions.

    -   $x-axis =$ average expression, in log counts per million (CPM)
    -   $y-axis =$ log fold change between conditions

-   **Volcano plot:** shows significantly differentially expressed genes appearing as points that are both statistically significant (located at the top) and have substantial fold changes (located on the left or right sides). This visualization enables identification of genes that are statistically and biologically significant.

    -   $x-axis =$ log fold change between conditions
    -   $y-axis =$ negative logarithm of the FDR-adjusted p-values

-   **Heatmap:** visualize gene expression patterns across different experimental conditions. Rows are genes, columns represent samples, and the colour intensity indicates the expression level of a gene in a specific sample. The genes are also clustered based on similar expression patterns, which provides insights into the overall structure and relationships within large datasets.

    -   These heatmaps illustrates the top 30 most significant DE genes
    
- **Venn diagram:** visualises the significant DE gene overlap between the previous RNA-seq experiment and the current. 

```{r deAnalysis}
fc = "none"
fdr = 0.1

comp <- colnames(contrast)

if (fc == "none") {
  lm_fit <- limma::lmFit(object = voom1, design = full_design) %>%
    limma::contrasts.fit(contrasts = contrast) %>% limma::eBayes()  
  lm_all <- lapply(1:ncol(lm_fit), function(y){limma::topTable(lm_fit, coef = y, number = Inf, adjust.method = "fdr")}) %>% setNames(.,comp)
  } else {
  lm_fit <- limma::lmFit(object = voom1, design = full_design) %>%
    limma::contrasts.fit(contrasts = contrast) %>% limma::treat(fc = as.numeric(fc))
  lm_all <- lapply(1:ncol(lm_fit), function(y){limma::topTreat(fit = lm_fit, coef = 1, number = Inf, adjust.method = "fdr")}) %>% setNames(.,comp)
}

lm_all <- lapply(lm_all, function(x) {
  df <- x %>% as.data.frame() %>% 
    dplyr::mutate(expression = case_when(
      adj.P.Val <= fdr & logFC >=0 ~ "up",
      adj.P.Val <= fdr & logFC <0 ~ "down",
      TRUE ~ "insig"))
  
  df$expression <- factor(df$expression, levels = c("insig", "down", "up"))
  df$entrezid <- df$entrezid %>% as.character()
  
  return(df)

})

lm_sig <- lapply(lm_all, function(x) {x %>% dplyr::filter(adj.P.Val <= fdr)})

```

## `r comp[1]` {.tabset .tabset-pills}

### P-val histogram

```{r hist}
lm_hist <- list()
for (name in comp) {
  lm_hist[[name]] <- ggplot(lm_all[[name]] %>% as.data.frame(), aes(x = P.Value)) +
    geom_histogram(bins = 50, fill="#8DA0CB",colour= "white", linewidth = 0.2, alpha=0.9) +
    scale_y_continuous(expand = expansion(mult = c(0, .1))) +
    labs(x = "P values", y = "Counts") +
    bossTheme(base_size = 14)
  
  if (savePlots == T) {
    ggsave(paste0("hist_",name,".svg"),
           plot = lm_hist[[name]],
           path = here::here("2_plots/2_DE/"),
           width = 13,
           height = 11,
           units = "cm")
  }
}

lm_hist[[1]]
```

### MA plot

```{r ma}
ma <- list()
for (name in comp) {
  top <- 5
  top_limma <- bind_rows(
    lm_all[[name]] %>%
      dplyr::filter(expression == "up") %>%
      arrange(adj.P.Val, desc(abs(logFC))) %>%
      head(top),
    lm_all[[name]] %>%
      dplyr::filter(expression == "down") %>%
      arrange(adj.P.Val, desc(abs(logFC))) %>%
      head(top)
  )
  invisible(top_limma %>% as.data.frame())
  
  ma[[name]] <- lm_all[[name]] %>% dplyr::arrange(expression) %>% 
    ggplot(aes(x = AveExpr, y = logFC)) +
    geom_point(aes(colour = expression, size = expression),show.legend = T, alpha = 0.7, stroke =0) +
    geom_label_repel(data = top_limma, # map labels, visit ?geom_label_repel
                     mapping = aes(label = gene),
                     size = 3,
                     label.padding = 0.15,
                     label.size = 0,
                     label.r = 0.15,
                     box.padding = 0.9,
                     point.padding = 0.5,
                     segment.size = 0.3,
                     segment.color = "grey50"
                     ) +
    labs(
      x = expression("log"[2] * "CPM"),
      y = expression("log"[2] * "FC"),
      colour = "Expression") + 
    {
      if (fc =="none") {
        geom_hline(yintercept = 0, linetype = "solid")
      } else {
        geom_hline(yintercept = c(-as.numeric(fc), 0, as.numeric(fc)), linetype = c("dashed", "solid", "dashed"))
      }
    } +
    scale_y_continuous(limits = c(-5,5), expand = expansion(mult = c(0, 0)), breaks = c(-6,-4,-2,0,2,4,6)) +
    scale_size_manual(values = c(1.5,2.4,2.4), guide = "none") +
    scale_fill_manual(values = expressionCol) +
    scale_color_manual(labels = c(paste0("NS: ", sum(lm_all[[name]]$expression == "insig"), "  "),
                                  paste0("Down: ", sum(lm_all[[name]]$expression == "down"), "  "),
                                  paste0("Up: ", sum(lm_all[[name]]$expression == "up"), " ")), 
                       values = expressionCol_dark) +    
    bossTheme(base_size = 14, legend = "bottom") +
    guides(colour = guide_legend(override.aes = list(size = 2.4)))
  
  
  if (savePlots) {
    ggsave(paste0("ma_", name, ".png"),
           plot = ma[[name]],
           path = here::here("2_plots/2_DE/"),
           width = 13,
           height = 11,
           units = "cm",
           dpi = 900)
  }
  
}

# display
ma[[1]]
```

### Volcano plot

```{r vol}
vol <- list()
for (name in comp) {
    top <- 5
    top_limma <- bind_rows(
      lm_all[[name]] %>%
        dplyr::filter(expression == "up") %>%
        arrange(adj.P.Val, desc(abs(logFC))) %>%
        head(top),
      lm_all[[name]] %>%
        dplyr::filter(expression == "down") %>%
        arrange(adj.P.Val, desc(abs(logFC))) %>%
        head(top)
    )
    invisible(top_limma %>% as.data.frame())
    
    vol[[name]] <- lm_all[[name]] %>%
      ggplot(aes(x = logFC,y = -log(adj.P.Val, 10))) +
      geom_point(aes(colour = expression, size = expression), show.legend = T, alpha =0.7, stroke =0 ) +
      geom_label_repel(data = top_limma, 
                       mapping = aes(label = gene),
                       size = 3,
                       label.padding = 0.15,
                       label.size = 0,
                       label.r = 0.15,
                       box.padding = 0.9,
                       point.padding = 0.5,
                       segment.size = 0.3,
                       segment.color = "grey50") +
      labs(x = expression("log"[2] * "FC"),
           y = expression("-log"[10] * "FDR"),
           colour = "Expression") +
      scale_x_continuous(limits = c(-5,5),expand = expansion(mult = c(0, 0)), breaks = c(-6,-4,-2,0,2,4,6)) +
      scale_size_manual(values = c(1.5,2.4,2.4), guide = "none") +
      scale_fill_manual(values = expressionCol) +
      scale_color_manual(labels = c(paste0("NS: ", sum(lm_all[[name]]$expression == "insig"), "  "),
                                    paste0("Down: ", sum(lm_all[[name]]$expression == "down"), "  "),
                                    paste0("Up: ", sum(lm_all[[name]]$expression == "up"), " ")), 
                         values = expressionCol_dark) +    
      bossTheme(base_size = 14, legend = "bottom") +
      guides(colour = guide_legend(override.aes = list(size = 2.4)))
    
    
    if (savePlots == T) { 
      ggsave(paste0("vol_", name, ".png"),
             plot = vol[[name]],
             path = here::here("2_plots/2_DE/"),
             width = 11,
             height = 13,
             units = "cm",
             dpi = 900)
    }
  
}

# display
vol[[1]]
```

### Heatmap

```{r hmap, fig.height=10, fig.width=8}
logCPM_sig=list()
anno=list()

for (i in 1:length(comp)) {
  name <- comp[i] %>% as.character()
    # create df with normalised read counts with an additional entrezid column for binding
  logCPM <- cpm(dge, prior.count = 2, log = TRUE)
  rownames(logCPM) <- dge$genes$gene
  
  # df for heatmap annotation of sample group
  anno[[name]] <- as.factor(dge$samples$group) %>% as.data.frame()
  colnames(anno[[name]]) <- "Groups"
  # rownames(anno) <- colnames(logCPM)
  
  if (i == 1) { # for the first comparison, extract just intact and control
    logCPM <- subset(logCPM ,select = c(6:11,1:5))
    anno[[name]] <- dplyr::slice(anno[[name]], c(6:11,1:5))
    rownames(anno[[name]]) <- colnames(logCPM)
    # anno_colours[[x]] <- anno_colours[[x]][1:2]
  } else if (i == 2) { # for the second comapsion, extract intact and SVX_VAS
    logCPM <- subset(logCPM, select = c(6:11,12:16))
    anno[[name]] <- dplyr::slice(anno[[name]], c(6:11,12:16))
    rownames(anno[[name]]) <- colnames(logCPM)
    # anno_colours[[x]] <- anno_colours[[x]][c(2,4)]
  } else if (i == 3) { # for the second comapsion, extract SVX and SVX_VAS
    logCPM <- subset(logCPM, select = c(1:5,12:16))
    anno[[name]] <- dplyr::slice(anno[[name]], c(1:5,12:16))
    rownames(anno[[name]]) <- colnames(logCPM)
    # anno_colours[[x]] <- anno_colours[[x]][3:4]
  } 
  
  # filtering top unregulated genes then filter the logCPM values of those genes.
  sig_de <- lm_sig[[name]] %>%
    arrange(sort(adj.P.Val, decreasing = F))
  sig_de <- sig_de[1:30,]
  logCPM_sig[[name]] <- logCPM[sig_de$gene,] %>% as.data.frame()
}

lm_hmap = list()
for (name in comp) {

  lm_hmap[[name]] <- ComplexHeatmap::pheatmap(
    mat = logCPM_sig[[name]],
    color = colorRampPalette(rev(c("#FB8072","#FDB462","#ffffd5","#8DD3C7","#80B1D3")))(100),
    scale = "row",
    cluster_cols = T,
    border_color = "white",
    cutree_cols = 2,
    cutree_rows = 4,
    treeheight_row = 40,
    treeheight_col = 30,
    show_colnames = F,
    clustering_distance_rows = "euclidean",
    main = paste0("Top ", nrow(logCPM_sig[[name]]), " significant DEGs"),


    legend = F,

    # heatmap_legend_param = list(title = "Z-score",
    #                             legend_direction = "vertical",
    #                             legend_width = unit(5, "cm")),

    # annotation = T,
    annotation_legend = T,
    annotation_col = anno[[name]],
    annotation_colors = list("Groups" = groupColour),
    annotation_names_col = F,

    fontfamily = "Arial Narrow",
    fontsize = 14,
    fontsize_col = 14,
    fontsize_number = 14,
    fontsize_row = 14,
    labels_row = as.expression(lapply(sig_de$gene, function(a) bquote(italic(.(a)))))
  ) %>% as.ggplot()
  
  if (savePlots == T) {
    ggsave(paste0("heat_",name,".svg"),
           plot = lm_hmap[[name]],
           path = here::here("2_plots/2_DE/"),
           width = 12,
           height = 16,
           units = "cm")
  }
}

lm_hmap[[1]]

```

```{r loop, fig.width=8, fig.height=8,results="asis"}
## this function is basically creating chunks within chunks, and then
## I use results='asis' so that the html image code is rendered 
kexpand <- function(wd, ht, cap) {
  cat(knit(text = knit_expand(text = 
     sprintf("```{r %s, fig.width=%s, fig.height=%s}\n.pl\n```", cap,wd, ht)
)))}

# Loop through each FC value

types <- c("P-val histogram", "MA plot", "Volcano plot", "Heatmap")

for (i in 2:length(comp)) {
  
  cat(paste0("## ",comp[i],"{.tabset .tabset-pills} \n\n"))

  cat(paste0("### ",types[[1]]," \n"))
  .pl <- lm_hist[[i]] 
  kexpand(wd = 10,ht = 6,cap = paste0("hist_",i))
  cat("\n\n")
  
  cat(paste0("### ",types[[2]]," \n"))
  .pl <- ma[[i]] 
  kexpand(wd = 10,ht = 6,cap = paste0("ma_",i))
  cat("\n\n")
  
  cat(paste0("### ",types[[3]]," \n"))
  .pl <- vol[[i]]
  kexpand(wd = 10,ht = 6,cap = paste0("vol_",i))
  cat("\n\n")
  
  cat(paste0("### ",types[[4]]," \n"))
  .pl <- lm_hmap[[i]] 
  kexpand(wd = 8,ht = 10,cap = paste0("hmap_",i))
  cat("\n\n")
}

```

## Combined {.tabset .tabset-pills}
```{r, eval=FALSE}
#create matrix with log cpm counts
logCPM <- cpm(dge, prior.count=2, log=TRUE)
rownames(logCPM) <- dge$genes$gene

#join common significant DE genes into df
common <- join_all(list(lm_sig[[1]], lm_sig[[2]], lm_sig[[3]]), by = 'gene', type = 'inner')
```

### Venn diagram

```{r vennDiagram, fig.height=8, fig.width=8, echo=FALSE,results='hide'}
futile.logger::flog.threshold(futile.logger::ERROR, name = "VennDiagramLogger")

# for(name in comp) {
  venn.diagram(x = list("DT vs PBS"   = lm_sig[[1]]$gene %>% as.character(), 
                        "Treg vs PBS" = lm_sig[[2]]$gene %>% as.character(),
                        "Treg vs DT"  = lm_sig[[3]]$gene %>% as.character()),
               filename = here::here(paste0("2_plots/2_DE/venn.png")),
               lwd = 2,
               fill = groupColour,
               alpha = 0.75,
               lty = 'blank',
               imagetype = "png",
               
               # Numbers
               cex = 2,
               fontface = "plain",
               fontfamily = "Arial Narrow",
               
               # Set names
               cat.cex = 2,
               cat.fontface = "bold",
               cat.fontfamily = "Arial Narrow",
               cat.default.pos = "outer",
               cat.pos = c(-27, 27, 135),
               cat.dist = c(0.055, 0.055, 0.085),
  )
 
```

![](figure/deAnalysis.Rmd/venn.png)

### Heatmap

```{r heat_combined, fig.width=9, fig.height=11}
logCPM_combined <- cpm(dge, prior.count=3, log=TRUE)
rownames(logCPM_combined) <- dge$genes$gene

#join common significant DE genes into df
common <- join_all(list(lm_sig[[1]], lm_sig[[3]]), by = 'gene', type = 'inner')

#merge the log cpm counts with the top 30 common de genes
logCPM_combined <- logCPM_combined[common$gene[1:nrow(common)],]
logCPM_combined <- logCPM_combined[,c(6,7,8,9,10,11,1,2,3,4,5,12,13,14,15,16)]

#df for heatmap annotation of sample type
anno_combined <- factor(dge$samples$group, levels = c("PBS", "DT", "Treg")) %>% as.data.frame()
anno_combined <- anno_combined[c(6,7,8,9,10,11,1,2,3,4,5,12,13,14,15,16),]%>% as.data.frame()

colnames(anno_combined) <- "Groups"


heat_combined <- ComplexHeatmap::pheatmap(
    mat = logCPM_combined,
    color = colorRampPalette(rev(c("#FB8072","#FDB462","#ffffd5","#8DD3C7","#80B1D3")))(300),
    scale = "row",
    cluster_cols = F,
    border_color = "white",
    gaps_col = c(6,11),
    # cutree_cols = 3,
    cutree_rows = 6,
    treeheight_row = 40,
    treeheight_col = 30,
    show_colnames = T,
    clustering_distance_rows = "euclidean",
    # main = paste0("Top ", nrow(logCPM_combined), " significant DEGs"),


    legend = T,

    heatmap_legend_param = list(title = "Expression\nZ-score",
                                direction= "vertical",
                                merge_legend = T,
                                legend_direction = "vertical",
                                legend_width = unit(5, "cm")),

    # annotation = T,
    annotation_legend = T,
    annotation_col = anno_combined,
    annotation_colors = list("Groups" = groupColour),
    annotation_names_col = T,
    # annotation_legend_param = list(direction = "horizontal"),

    fontfamily = "Arial Narrow",
    fontsize = 12,
    fontsize_col = 12,
    fontsize_number = 12,
    fontsize_row = 12,
    labels_row = as.expression(lapply(rownames(logCPM_combined), function(a) bquote(italic(.(a)))))
  )

draw(heat_combined, merge_legend = T, heatmap_legend_side = "right", 
    annotation_legend_side = "right")

if (savePlots == T) {
  svg(filename = here::here("2_plots/2_DE/heat_combined.svg"),width = 8,height = 12)
  draw(heat_combined, merge_legend = T, heatmap_legend_side = "bottom", 
       annotation_legend_side = "bottom")
  dev.off()
  }

```


# Export Data

The following are exported:

-   **de_genes_all.xlsx** - This spreadsheet contains all DE genes. 

-   **de_genes_sig.xlsx** - This spreadsheet contains only significant DE genes.

```{r exportData, eval=export}
lapply(lm_all, function(x) x %>% dplyr::select(c("gene", "logFC", "AveExpr", "P.Value", "adj.P.Val", "description", "entrezid","expression"))) %>% writexl::write_xlsx(x = ., here::here("3_output/de_genes_all.xlsx"))

lapply(lm_sig, function(x) x %>% dplyr::select(c("gene", "logFC", "AveExpr", "P.Value", "adj.P.Val", "description", "entrezid","expression"))) %>% writexl::write_xlsx(x = ., here::here("3_output/de_genes_sig.xlsx"))

# lapply(common, function(x) x %>% dplyr::select(c("gene", "logFC", "AveExpr", "P.Value", "adj.P.Val", "description", "entrezid","expression", "Type"))) %>% writexl::write_xlsx(x = ., here::here("3_output/common_deg.xlsx"))
```

```{r saveRDS, include=FALSE}
saveRDS(object = lm, file = here::here("0_data/rds_objects/lm.rds"))

saveRDS(object = lm_all, file = here::here("0_data/rds_objects/lm_all.rds"))

saveRDS(object = lm_sig, file = here::here("0_data/rds_objects/lm_sig.rds"))

saveRDS(comp, here::here("0_data/rds_objects/comp.rds"))
```
