---
title: "Gene ontology (GO) analysis"
author: "Ha M. Tran"
date: "08-01-2024"
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  echo = TRUE,
  message = FALSE,
  warning = FALSE,
  eval = TRUE,
  fig.align = "center"
)
```


```{r load libraries}
# working with data
library(dplyr)
library(magrittr)
library(readr)
library(tibble)
library(reshape2)
library(tidyverse)

# Visualisation:
library(kableExtra)
library(ggplot2)
library(grid)
library(DT)
library(extrafont)
library(VennDiagram)
# Custom ggplot
library(gridExtra)
library(ggbiplot)
library(ggrepel)
library(rrvgo)
library(d3treeR)
library(plotly)
library(GOSemSim)
library(data.table)

# Bioconductor packages:
library(edgeR)
library(limma)
library(Glimma)
library(clusterProfiler)
library(org.Mm.eg.db)
library(enrichplot)
library(patchwork)

library(pandoc)
library(knitr)
opts_knit$set(progress = FALSE, verbose = FALSE)
opts_chunk$set(warning=FALSE, message=FALSE, echo=FALSE)

```

```{r importData}
# load DGElist previously created in the set up
dge <- readRDS(here::here("0_data/rds_objects/dge.rds"))
lm <- readRDS(here::here("0_data/rds_objects/lm.rds"))
lm_sig <- readRDS(here::here("0_data/rds_objects/lm_sig.rds"))
Comp <- readRDS(here::here("0_data/rds_objects/comp.rds"))
Ont <- c("BP","MF","CC")
# to increase the knitting speed. change to T to save all plots
savePlots <- F
export <- F
```

```{r importFunctions}
# Theme
bossTheme <- readRDS(here::here("0_data/functions/bossTheme.rds"))
bossTheme_bar <- readRDS(here::here("0_data/functions/bossTheme.rds"))
groupColour <- readRDS(here::here("0_data/functions/groupColour.rds"))
groupColour_dark <- readRDS(here::here("0_data/functions/groupColour_dark.rds"))
expressionCol <- readRDS(here::here("0_data/functions/expressionCol.rds"))
expressionCol_dark <- readRDS(here::here("0_data/functions/expressionCol_dark.rds"))


DT <- readRDS(here::here("0_data/functions/DT.rds"))

# Plotting
convert_to_superscript <- readRDS(here::here("0_data/functions/convert_to_superscript.rds"))
exponent <- readRDS(here::here("0_data/functions/exponent.rds"))
format_y_axis <- readRDS(here::here("0_data/functions/format_y_axis.rds"))

firstCap <- function(x) {
  substr(x, 1, 1) <- toupper(substr(x, 1, 1))
  x
}

```

# Gene ontology (GO) Analysis 

Functional enrichment analysis is a method used to identify biological functions or processes overrepresented in a set of genes or proteins. 

Gene Ontology (GO) is a standardized system for annotating genes and their products with terms from a controlled vocabulary, organized into three main categories: Molecular Function, Biological Process, and Cellular Component.  

-   **Biological Process (BP):** Describes the larger, coordinated biological events or processes in which a gene product is involved. This category represents a series of molecular events that contribute to a specific function.

-   **Molecular Function (MF):** Describes the specific molecular activities that a gene product performs, such as catalytic or binding activities.

-   **Cellular Component (CC):** Describes the location or structure within the cell where a gene product is active, such as the nucleus, cytoplasm, or membrane.

Each of these three main categories is further organized into a hierarchical structure with more specific terms. The terms become more specialized as you move down the hierarchy (ontology level). Comparing a gene list to a reference database offers critical insights into the biological significance of gene expression changes.

```{r}
# circumvent rerunning of lengthy analysis.
enrichGO <- readRDS(here::here("0_data/rds_objects/enrichGO.rds"))
enrichGO_sig <- readRDS(here::here("0_data/rds_objects/enrichGO_sig.rds"))
```

```{r enrichGO, eval=FALSE}
# `goSummaries` is a package created by Dr Stevie Pederson for filtering GO terms based on ontology level.
goSummaries <- url("https://uofabioinformaticshub.github.io/summaries2GO/data/goSummaries.RDS") %>%
  readRDS()
minPath <- 3

mmGO <- lapply(c("BP","MF","CC"), function(ont){godata('org.Mm.eg.db', ont=ont)}) %>% setNames(c("BP","MF","CC"))

enrichGO=list()
enrichGO_sig <- list()

for (comp in Comp) {
  # find enriched GO terms
  enrichGO[[comp]] <- clusterProfiler::enrichGO(
    gene =lm_sig[[comp]]$entrezid,
    universe = lm_all[[comp]]$entrezid,
    OrgDb = org.Mm.eg.db,
    keyType = "ENTREZID",
    ont = "ALL",
    pAdjustMethod = "fdr",
    pvalueCutoff = 0.05
  )
  
  enrichGO[[comp]] <- pairwise_termsim(enrichGO[[comp]], method = "Wang",semData = mmGO[[1]], showCategory = nrow(enrichGO[[comp]]@result))
  
}

for (comp in Comp) {
  # bind to goSummaries to elminate go terms with ontology levels 1 and 2.
  enrichGO_sig[[comp]] <- enrichGO[[comp]] %>% 
    clusterProfiler::setReadable(OrgDb = org.Mm.eg.db, keyType = "auto")
  
  enrichGO_sig[[comp]] <- enrichGO_sig[[comp]] %>%
    as.data.frame() %>%
    rownames_to_column("id") %>%
    left_join(goSummaries) %>%
    # dplyr::filter(shortest_path >= minPath) %>%
    column_to_rownames("id")
  
  # adjust go results, separate compound column, add FDR column, adjust the GeneRatio column
  enrichGO_sig[[comp]] <- enrichGO_sig[[comp]] %>%
    separate(col = BgRatio, sep = "/", into = c("Total", "Universe")) %>%
    dplyr::mutate(
      logFDR = -log(p.adjust, 10),
      GeneRatio = Count / as.numeric(Total))
  # %>% 
  #   dplyr::select(c("Description", "ontology", "GeneRatio", "pvalue", "p.adjust", "logFDR", "qvalue", "geneID", "Count"))
  
  
  enrichGO_sig[[comp]]$Description <- enrichGO_sig[[comp]]$Description %>% firstCap() %>% str_wrap(width = 45)
  
  
}
```

```{r semanticSim, eval=FALSE}
simMatrix <- list()
scores <- list()
reducedTerms <- list()


for(ont in Ont) {
  
  simMatrix[[ont]] <- lapply(Comp, function(comp) {
    calculateSimMatrix(enrichGO[[comp]]@result$ID,
                       orgdb="org.Mm.eg.db",
                       ont=ont,
                       method="Wang",
                       semdata = mmGO[[ont]])
  }) %>% setNames(Comp)
  
  scores[[ont]] <- lapply(enrichGO, function(x) {setNames(-log10(x@result$p.adjust), x@result$ID)}) %>% setNames(names(enrichGO))
  
  reducedTerms[[ont]] <- lapply(Comp, function(comp) {
    reduced_tb <- reduceSimMatrix(simMatrix[[ont]][[comp]],
                                    scores[[ont]][[comp]],
                                    threshold=0.7,
                                    orgdb="org.Mm.eg.db")

    reduced_tb$parentTerm <- reduced_tb$parentTerm %>% firstCap() %>% str_wrap(width = 45)
    reduced_tb$term <- reduced_tb$term %>% firstCap() %>% str_wrap(width = 35)

    return(reduced_tb)
  }) %>% setNames(Comp)
  
}



semSim_df <- list()
reduced_semSim_df <- list()
for(ont in Ont) {
  for (comp in Comp) {
    subset_df <- subset(reducedTerms[[ont]][[comp]], parent == rownames(reducedTerms[[ont]][[comp]]))
    parentTerm_size <- reducedTerms[[ont]][[comp]]$parentTerm %>% as.factor() %>% summary(500)
    semSim_df[[ont]][[comp]] <- cbind(subset_df, parentTerm_size)
  }
  
  reduced_semSim_df[[ont]] <- as.data.frame(do.call(rbind, semSim_df[[ont]])) %>%
    rownames_to_column("comparison")
  reduced_semSim_df[[ont]]$comparison <- gsub(pattern = "\\..*", "", reduced_semSim_df[[ont]]$comparison) %>% as.factor()
}

reducedTerms_all <- as.data.frame(do.call(rbind, reduced_semSim_df)) %>%
  rownames_to_column("ont")

reducedTerms_all$ont <- gsub(pattern = "\\..*", "", reducedTerms_all$ont) %>% as.factor()


saveRDS(simMatrix, here::here("0_data/rds_objects/simMatrix_ora.rds"))
saveRDS(scores, here::here("0_data/rds_objects/scores_ora.rds"))
saveRDS(reducedTerms, here::here("0_data/rds_objects/reducedTerms_ora.rds"))
saveRDS(semSim_df, here::here("0_data/rds_objects/semSim_df.rds"))
saveRDS(reduced_semSim_df, here::here("0_data/rds_objects/reduced_semSim_df.rds"))
saveRDS(reducedTerms_all, here::here("0_data/rds_objects/reducedTerms_all.rds"))
```

## Visualisations {.tabset .tabset-pills}

The following visualisations are GO enrichment analysis performed with set of DE genes significantly below **FDR 0.1**  without FC threshold (TREAT). IMPORTANTLY, these GO terms are all significantly enriched (FDR <0.05)

-   **Dot plot:** illustrates the top 25 enriched GO terms.

    -   $Gene ratio =$ the number of significant DE gene in the term / the total of number of genes in the term. Indicated by the size
    -   The shapes represents the three main GO categories, either BP, MP, or CC

-   **Table:** list of all the significant GO terms

-   **Upset:** illustrate the overlap of gene between different functional terms

**Semantic similarity plots - GO specific**

Due to the hierarchical structure of Gene Ontologies, the enriched sets generated often exhibit redundancy and pose challenges in interpretation.
The subsequent analyses and visualizations seek to alleviate this redundancy in GO sets by grouping comparable terms based on their semantic similarity.
The underlying concept behind measuring semantic similarity is grounded in the idea that genes sharing similar functions should possess analogous annotation vocabulary and exhibit close relationships within the ontology structure.

NOTE: the following semantic similarity analyses are performed using Graph-based method (Wang et al. 2007)

-   **Dendrogram plot:** performs hierarchical clustering on the semantic similarity of GO terms.

    -   NOTE: to maintain readability, only the top 50 most significant GO terms are clustered. These clusters are then divided into 9 clades and labeled using the top 4 high-frequency words.

-   **Scatter plot:** illustrates the UMAP space between semantically similar significant GO terms

    -   Distances represent the similarity between terms,
    -   Size represents the significance (in $-\log_{10}FDR$))
    -   NOTE: to maintain reability, only the top 15 most significant parent terms are labeled. Parent terms are the most significant term in a particular cluster

-   **Treemap plot:** Visualise the of hierarchical structures of semantically similar GO terms.

    -   The terms are colored based on their parent term, 
    -   The size of the term is proportional to the significance.


**I recommend reading through the full list of significant GO terms and selecting the most biologically relevant for better visualisation**


### `r Comp[1]` {.tabset .tabset-pills}

#### Dot plot
```{r dot_og, fig.width=8, fig.height=10}
dot <- list()
tab <- list()
upset <- list()

for (comp in Comp) {
  dot[[comp]] <- ggplot(enrichGO_sig[[comp]][1:20, ]) +
    geom_point(aes(x = GeneRatio, y = reorder(Description, GeneRatio), colour = logFDR, size = Count, shape = ontology %>% as.factor())) +
    scale_color_gradientn(colors = rev(c("#FB8072","#FDB462","#ffffd5","#8DD3C7","#80B1D3")),
                          values = scales::rescale(c(min(enrichGO_sig[[comp]]$logFDR), max(enrichGO_sig[[comp]]$logFDR))),
                          breaks = scales::pretty_breaks(n = 5)) +
    scale_size(range = c(2,5)) +
    labs(x = "Gene ratio", y = "", color = expression("-log"[10] * "FDR"), size = "Gene Counts", shape = "Ontology")+
    bossTheme(base_size = 14,legend = "right")
  
  tab[[comp]] <- enrichGO_sig[[comp]] %>% 
    dplyr::mutate_if(is.numeric, funs(as.character(signif(.,3)))) %>% 
    DT(., caption = "Significantly enriched GO terms")
  
  
  upset[[comp]] <- upsetplot(x = enrichGO[[comp]], 10)
  
  if(savePlots == TRUE) {
    ggsave(filename = paste0("dot_", comp, ".svg"), plot = dot[[comp]], path = here::here("2_plots/3_FA/go/"),
           width = 18, height = 20, units = "cm")
    ggsave(filename = paste0("upset_", comp, ".svg"), plot = upset[[comp]], path = here::here("2_plots/3_FA/go/"), width = 20, height = 14, units = "cm")
    
  }
}

dot[[1]]
```

#### Table
```{r tab}
tab[[1]]
```

#### Upset plot

```{r upset, fig.width=11, fig.height=9}
upset[[1]]
```


#### Dendrogram

```{r semSim_den, fig.width=8, fig.height=11}
den <- lapply(enrichGO, function(x) {
  treeplot(x,showCategory = 50, fontsize = 3, cex_category = 0.5,
         cluster.params      = list(method = "ward.D", n = 9, label_words_n = 4, label_format = 30),
         hilight.params      = list(hilight = F, align = "both"),
         clusterPanel.params = list(clusterPanel = "heatMap", pie = "equal", legend_n = 3),
         offset.params       = list(bar_tree = rel(3), tiplab = rel(4), extend = 0.1, hexpand = 0.4)) +
  theme(legend.position = "bottom")
}) %>% setNames(enrichGO)

# saveRDS(den, here::here("0_data/rds_objects/ora_dendrogram.rds"))
# den <- readRDS(here::here("0_data/rds_objects/ora_dendrogram.rds"))

den[[1]]
```

#### Scatter plot

```{r semSim_scatter, fig.width = 8, fig.height=8}
simMatrix <- readRDS(here::here("0_data/rds_objects/simMatrix_ora.rds"))
scores <- readRDS(here::here("0_data/rds_objects/scores_ora.rds"))
reducedTerms <- readRDS(here::here("0_data/rds_objects/reducedTerms_ora.rds"))

revigo_dimReduction <- function(simMatrix, reducedTerms, algorithm = c("pca", "umap"), size = "score") {
  
  x <- switch(match.arg(algorithm), pca = cmdscale(as.matrix(as.dist(1 - simMatrix)), eig = TRUE, k = 3)$points, umap = umap::umap(as.matrix(as.dist(1 - simMatrix)),n_components = 3)$layout)
  df <- cbind(as.data.frame(x), reducedTerms[match(rownames(x), reducedTerms$go), c("term", "parent", "parentTerm", size)])
  
  return(df)
}
dimReduced <- list()
scatter <- list()
for (comp in Comp){
  set.seed(3)
  dimReduced[[comp]] <- revigo_dimReduction(simMatrix[["BP"]][[comp]], reducedTerms[["BP"]][[comp]], algorithm = "umap")
  
  m <- subset(dimReduced[[comp]], parent == rownames(dimReduced[[comp]]))[1:15,]
  
  scatter[[comp]] <- ggplot(dimReduced[[comp]], aes(x = V1, y = V2, color = parentTerm)) +
    geom_point(aes_string(size = "score"), alpha = 0.5, stroke = 0) + 
    scale_color_discrete(guide = "none") +
    # scale_fill_discrete(guide = "none") +
    scale_size_continuous(name = expression("-log"[10] * "FDR"), range = c(0,12)) + 
    scale_x_continuous(name = "UMAP1") +
    scale_y_continuous(name = "UMAP2") +
    guides(x = ggh4x::guide_axis_truncated(trunc_lower = unit(0, "npc"),trunc_upper = unit(3, "cm")), 
           y = ggh4x::guide_axis_truncated(trunc_lower = unit(0, "npc"),trunc_upper = unit(3, "cm"))) +
    geom_label_repel(aes(label = m$parentTerm),data = m, box.padding = grid::unit(1,"lines"), size = 3, label.size = 0.15) + 
    bossTheme(14) +
    theme(
      line = element_blank(),
      rect = element_blank(),
      panel.border = element_blank(),
      axis.text.x = element_blank(),
      axis.text.y = element_blank(),
      axis.ticks = element_blank(),
      legend.position = "right",
      axis.line = element_line(arrow = arrow()),
      axis.title = element_text(hjust = 0),
      axis.title.y = element_text(
        family = "Arial Narrow",
        face = "plain",
        size = 13,
        angle = 90,
        vjust = 1
      ),
      axis.title.x = element_text(
        family = "Arial Narrow",
        face = "plain",
        size = 13,
        angle = 0,
        vjust = 0
      )
    )
  
}

scatter[[1]]
```

Interactive scatter

```{r interactiveScatter, fig.width = 9, fig.height=9}
scatter_plotly <- list()
for (comp in Comp){
  scatter_plotly[[comp]] <- ggplotly(scatter[[comp]] + bossTheme(14)) %>% add_markers(size = 5) %>% layout(showlegend = FALSE)
}

scatter_plotly[[1]]
```


3D Interactive scatter


```{r 3d interactive scatter, fig.width = 9, fig.height=9}
scatter_3d <- list()

for(comp in Comp){
  scatter_3d[[comp]] <- plot_ly(dimReduced[[comp]], x = ~ V1, y = ~ V2, z = ~ V3, color = ~ parentTerm, size = ~ score,
  marker = list(symbol = 'circle', sizemode = 'diameter'),
  sizes = c(5, 70),
  text = ~paste('Term    :', term,'<br>P. Term:', parentTerm, '<br>Sig       :', score),
  hoverinfo = 'text') %>% 
  layout(showlegend = FALSE,
         title = 'Semantically similar GO terms',
         scene = list(xaxis = list(title = 'UMAP 1',
                      gridcolor = 'rgb(255, 255, 255)',
                      zerolinewidth = 1,
                      ticklen = 5,
                      gridwidth = 2),
               yaxis = list(title = 'UMAP 2',
                      gridcolor = 'rgb(255, 255, 255)',
                      zerolinewidth = 1,
                      ticklen = 5,
                      gridwith = 2),
               zaxis = list(title = 'UMAP 3',
                            gridcolor = 'rgb(255, 255, 255)',
                            zerolinewidth = 1,
                            ticklen = 5,
                            gridwith = 2)))
  
}

scatter_3d[[1]]
```

#### Parent terms

```{r parentTerms, fig.width=9, fig.height=14}
semSim_df <- readRDS(here::here("0_data/rds_objects/semSim_df.rds"))

semSim_dot <- lapply(Comp, function(comp){
  
  df <- semSim_df[["BP"]][[comp]]
  df$parentTerm <- df$parentTerm %>% str_wrap(50)
  
  plot <- ggplot(df) +
    geom_point(aes(x = parentTerm_size, y = reorder(parentTerm, score), colour = score, size = parentTerm_size)) +
    scale_color_gradientn(colors = rev(c("#FB8072","#FDB462","#ffffd5","#8DD3C7","#80B1D3")),
                          values = scales::rescale(c(min(df$score), max(df$score))),
                          breaks = scales::pretty_breaks(n = 5)) +
    scale_size(range = c(2,5), guide = F) +
    labs(x = "Term size", y = "", color = expression("-log"[10] * "FDR"))+
    bossTheme(base_size = 14,legend = "right")
  
  if(savePlots == TRUE) {
    ggsave(filename = paste0("parTerm_dot_", comp, ".svg"), plot = plot, path = here::here("2_plots/3_FA/go/"),
           width = 18, height = 20, units = "cm")
  }
  
  return(plot)
  
}) 


semSim_dot[1]
```


#### Treemap plot

```{r semSim_tree, fig.width=8, fig.height=8}
adjTreemap <- function (x, size = "score", title = "", ...) {
  treemap::treemap(
    x,
    index = c("parentTerm", "term"),
    vSize = size,
    type = "index",
    title = title,
    palette = "Set2",
    fontcolor.labels = c("grey85","#00000080"),
    bg.labels = 0,
    border.col = "grey10",
    border.lwds = c(1,0.5),
    fontfamily.labels = "Arial Narrow"
  )
}

adjTreemap(reducedTerms[[1]][[1]])

```

Interactive Tree

```{r interactiveTree, fig.width=9,fig.height=9}
# png(filename=here::here("2_plots/tree.png"),width=20, height=20,units = "cm",res = 900)
# tree <- adjTreemap(reducedTerms)
# dev.off()

library(htmltools)

interactive_treemap <- function(x){
  browsable(
    tagList(
      tags$head(
        tags$style('text.label{font-size: 20px !important}')
      ),
      d3tree3(adjTreemap(x), rootname = "General")
    )
  )
}

interactive_treemap(reducedTerms[[1]][[1]])
```

```{r loop, results="asis"}
## this function is basically creating chunks within chunks, and then
## I use results='asis' so that the html image code is rendered 
kexpand <- function(wd, ht, cap, res, echo) {
  cat(knit(text = knit_expand(text = 
     sprintf("```{r %s, results='%s', echo = '%s',fig.keep='all', fig.width=%s, fig.height=%s}\n.pl\n```", cap, res, echo, wd, ht)
)))}

special_kexpand <- function(wd, ht, cap, res, echo) {
  cat(knit(text = knit_expand(text = 
     sprintf("```{r %s, results='%s', echo = '%s',fig.keep='all', fig.width=%s, fig.height=%s}\ninteractive_treemap(reducedTerms[[1]][[i]])\n```", cap, res, echo, wd, ht)
)))}

# Loop through each FC value
headers <- Comp
types <- c("Dot plot", "Table", "Upset plot", "Dendrogram", "Scatter", "Parent term", "Treemap")

for (i in 2:length(headers)) {
  cat(paste0("### ",headers[i],"{.tabset .tabset-pills} \n\n"))
  
  cat(paste0("#### ",types[[1]]," \n"))
  .pl <- dot[[i]] 
  kexpand(wd = 8,ht = 10,cap = paste0("dot",i),res = "markup",echo = "TRUE")
  cat("\n\n")
  
  cat(paste0("#### ",types[[2]]," \n"))
  .pl <- tab[[i]]
  kexpand(wd = 11,ht = 8,cap = paste0("tab",i), res = "markup",echo = "TRUE")
  cat("\n\n")
  
  cat(paste0("#### ",types[[3]]," \n"))
  .pl <- upset[[i]] 
  kexpand(wd = 11,ht = 9,cap = paste0("upset",i),res = "markup",echo = "TRUE")
  cat("\n\n")
  
  cat(paste0("#### ",types[[4]]," \n"))
  .pl <- den[[i]] 
  kexpand(wd = 8,ht = 11,cap = paste0("den",i),res = "markup",echo = "TRUE")
  cat("\n\n")
  
  cat(paste0("#### ",types[[5]]," \n"))
  .pl <- scatter[[i]] 
  kexpand(wd = 8,ht = 8,cap = paste0("scatter",i),res = "markup",echo = "TRUE")
  cat("\n\n")
  
  cat(paste0("Interactive Scatter \n"))
  .pl <- scatter_plotly[[i]] 
  kexpand(wd = 9,ht = 9,cap = paste0("scatter_interactive",i),res = "markup",echo = "TRUE")
  cat("\n\n")
  
  cat(paste0("3D scatter \n"))
  .pl <- scatter_3d[[i]] 
  kexpand(wd = 9,ht = 9,cap = paste0("scatter_3d",i),res = "markup",echo = "TRUE")
  cat("\n\n")
  
  cat(paste0("#### ",types[[6]]," \n"))
  .pl <- semSim_dot[[i]]
  kexpand(wd = 9,ht = 12 ,cap = paste0("parentTerm",i),res = "markup",echo = "TRUE")
  cat("\n\n")
  
  cat(paste0("#### ",types[[7]]," \n"))
  # special_kexpand(wd = 9,ht = 9,cap = paste0("treemap",i),res = "hide",echo = "FALSE")
  # cat("\n\n")
  
  # cat(paste0("Interactive Treemap\n"))
  # .pl <- interactive_treemap(reducedTerms[[i]])
  special_kexpand(wd = 9,ht = 9,cap = paste0("treemap",i),res = "markup",echo = "FALSE")
  cat("\n\n")
}
```



### Combined {.tabset .tabset-pills}

#### Venn diagram

```{r combined_venn, fig.height=8, fig.width=8, echo=FALSE, results='hide'}
futile.logger::flog.threshold(futile.logger::ERROR, name = "VennDiagramLogger")

venn.diagram(x = list("DT vs PBS"   = enrichGO_sig[[1]]$ID %>% as.character(), 
                      "Treg vs PBS" = enrichGO_sig[[2]]$ID %>% as.character(),
                      "Treg vs DT"  = enrichGO_sig[[3]]$ID %>% as.character()),
             filename = here::here(paste0("2_plots/3_FA/go/venn.png")),
             lwd = 2,
             disable.logging = T,
             fill = groupColour,
             alpha = 0.75,
             lty = 'blank',
             imagetype = "png",
             
             # Numbers
             cex = 2,
             fontface = "plain",
             fontfamily = "Arial Narrow",
             
             # Set names
             cat.cex = 2,
             cat.fontface = "bold",
             cat.fontfamily = "Arial Narrow",
             cat.default.pos = "outer",
             cat.pos = c(-27, 27, 135),
             cat.dist = c(0.055, 0.055, 0.085)
)

```

![](figure/go.Rmd/venn.png)


#### Dot plot

```{r combined_dot, fig.width=9, fig.height=13}
# combine all df in list into one df
go_dot_all <- as.data.frame(do.call(rbind, enrichGO_sig)) %>%
  rownames_to_column("group")

# clean group names and change to factor
go_dot_all$group <- gsub(pattern = "\\..*", "", go_dot_all$group) %>% as.factor()

# factor the descriptions
top10 <- as.data.frame(do.call(rbind, lapply(enrichGO_sig, "[", 1:15,3))) %>% rownames_to_column("group")
top10 <- melt(top10, "group")
terms <- top10$value %>% as.factor() %>% levels()
go_dot_all <- go_dot_all[go_dot_all$Description %in% terms,]
go_dot_all$group <- factor(go_dot_all$group,levels = c("DT vs PBS", "Treg vs PBS", "Treg vs DT" ))
go_dot_all$Description <- go_dot_all$Description %>% str_wrap(55)
combine_go <- ggplot(go_dot_all) +
    geom_point(aes(x = group, y = reorder(Description, logFDR), colour = logFDR, size = Count, shape = ONTOLOGY %>% as.factor())) +
    scale_color_gradientn(colors = rev(c("#FB8072","#FDB462","#ffffd5","#8DD3C7","#80B1D3")),
                          values = scales::rescale(c(min(go_dot_all$logFDR), max(go_dot_all$logFDR))),
                          breaks = scales::pretty_breaks(n = 5)) +
    scale_size(range = c(2,5)) +
    labs(x = "", y = "", color = expression("-log"[10] * "FDR"), size = "Counts", shape = "Ontology")+
    bossTheme(base_size = 14,legend = "right")
  
  
  if(savePlots == TRUE) {
    ggsave(filename = paste0("combine_go_dot.svg"), plot = combine_go, path = here::here("2_plots/3_FA/go/"),
           width = 18, height = 20, units = "cm")
  }

combine_go
```


#### Parent term Venn

```{r parTermVenn}
 # cat.pos = c(-27, 27, 135),
 #               cat.dist = c(0.055, 0.055, 0.085)
futile.logger::flog.threshold(futile.logger::ERROR, name = "VennDiagramLogger")
cat.pos <- list(BP =c(-15,180,15),
                MF =c(-170,170,0),
                CC =c(-27, 27, 135))
cat.dist <- list(BP =c(0.05, .09, .045),
                 MF =c(0.085, 0.06, 0.085),
                 CC =c(0.055, 0.055, 0.085))
for(ont in Ont){
  venn.diagram(x = list("DT vs PBS"   = reducedTerms[[ont]][[1]]$parentTerm %>% as.character(), 
                        "Treg vs PBS" = reducedTerms[[ont]][[2]]$parentTerm %>% as.character(),
                        "Treg vs DT"  = reducedTerms[[ont]][[3]]$parentTerm %>% as.character()),
               filename = here::here(paste0("2_plots/3_FA/go/parTerm_",ont,"_venn.png")),
               lwd = 2,
               disable.logging = T,
               fill = groupColour,
               alpha = 0.75,
               lty = 'blank',
               imagetype = "png",
               # main = paste0(ont, "parent terms"),
               
               # Numbers
               cex = 2,
               fontface = "plain",
               fontfamily = "Arial Narrow",
               
               # Set names
               cat.cex = 2,
               cat.fontface = "bold",
               cat.fontfamily = "Arial Narrow",
               cat.default.pos = "outer",
               cat.pos = cat.pos[[ont]],
               cat.dist = cat.dist[[ont]]
  )
}

```

![Biological Processes parent terms](figure/go.Rmd/parTerm_BP_venn.png)

![Molecular Functions parent terms](figure/go.Rmd/parTerm_MF_venn.png)

![Cellular Components parent terms](figure/go.Rmd/parTerm_CC_venn.png)

#### Parent term dot
```{r combined_parent, fig.width=9, fig.height=14, fig.cap="Biological processes parent terms"}
reducedTerms_all <- readRDS(here::here("0_data/rds_objects/reducedTerms_all.rds"))
reducedTerms_all_BP <- reducedTerms_all[!reducedTerms_all$ont %in% c("MF", "CC"),] %>% dplyr::arrange(desc(score))
reducedTerms_all_BP$parentTerm <- reducedTerms_all_BP$parentTerm %>% str_wrap(45)

dotplot <- function(data){
  ggplot(data) +
    geom_point(aes(x = comparison, y = reorder(parentTerm, score), colour = score, size = parentTerm_size), alpha=0.8) +
    scale_color_gradientn(colours = rev(c("#FB8072","#FDB462","#ffffd5","#8DD3C7","#80B1D3")),
                          limits = c(min(reducedTerms_all_BP$score), max(reducedTerms_all_BP$score)),
                          breaks = scales::pretty_breaks(n = 5)) +
    scale_size(range = c(2,7),limits = c(min(reducedTerms_all_BP$parentTerm_size), max(reducedTerms_all_BP$parentTerm_size))) +
    labs(x = "", y = "", color = expression("-log"[10] * "FDR"), size = "Term size")+
    bossTheme(base_size = 14,legend = "bottom")
}


t <- wrap_plots(list(dotplot(reducedTerms_all_BP[1:42, ]), dotplot(reducedTerms_all_BP[42:84, ]))) + 
  plot_layout(guides = "collect") &
  bossTheme(base_size = 12, legend = "bottom") &
  theme(legend.box.margin = margin(-5, 0, 0, 0, unit = "mm"),
        # plot.margin = margin(1, 1, 1, 0, unit = "pt"),
        axis.text.y = element_text(family = "Arial narrow", face = "plain", size = 9, hjust = 1),
        axis.text.x = element_text(family =  "Arial", face = "plain", size = 10, hjust = 1, vjust = 0.8, angle = 30))

t

ggsave(filename = "parentTerm_all.svg", plot = t, path = here::here("2_plots/3_FA/go/"),
           width = 21, height = 29.7, units = "cm")
```

# Export Data

The following are exported:

-   **GO.xlsx** - This spreadsheet contains all significantly enriched GO terms. NOTE:

```{r exportData, eval=export}

# save to excel
writexl::write_xlsx(x = enrichGO_sig, here::here("3_output/GO_sig.xlsx"))
saveRDS(object = enrichGO_sig,file = here::here("0_data/rds_objects/enrichGO_sig.rds"))
saveRDS(object = enrichGO,file = here::here("0_data/rds_objects/enrichGO.rds"))

```
