<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Ha Tran" />

<meta name="date" content="2024-01-12" />

<title>Extra figures</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.33/datatables.js"></script>
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<script src="site_libs/jszip-1.13.6/jszip.min.js"></script>
<script src="site_libs/pdfmake-1.13.6/pdfmake.js"></script>
<script src="site_libs/pdfmake-1.13.6/vfs_fonts.js"></script>
<link href="site_libs/dt-ext-buttons-1.13.6/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.13.6/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.6/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.6/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.6/js/buttons.print.min.js"></script>
<link href="site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Treg-uNK</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="setUp.html">Pre-processing</a>
</li>
<li>
  <a href="deAnalysis.html">DE Analysis</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Over-representation Analysis (ORA)
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="go.html">Gene Ontology</a>
    </li>
    <li>
      <a href="kegg.html">KEGG</a>
    </li>
    <li>
      <a href="reactome.html">Reactome</a>
    </li>
    <li>
      <a href="ipa.html">IPA</a>
    </li>
  </ul>
</li>
<li>
  <a href="extraFigs.html">Extra figures</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://www.linkedin.com/in/ha-tran-6973b5130/">
    <span class="fa fa-linkedin"></span>
     
  </a>
</li>
<li>
  <a href="https://github.com/tranmanhha135/">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Extra figures</h1>
<h4 class="author">Ha Tran</h4>
<h4 class="date">2024-01-12</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-08-02
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>5_Treg_uNK/1_analysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed12345code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(12345)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed12345code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(12345)</code> was run prior to running the
code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtranmanhha135TreguNKtreea5cdd4ec7e940249cd3f0b44b2c6e7aa34ccaa26targetblanka5cdd4ea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/tranmanhha135/Treg_uNK/tree/a5cdd4ec7e940249cd3f0b44b2c6e7aa34ccaa26" target="_blank">a5cdd4e</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomtranmanhha135TreguNKtreea5cdd4ec7e940249cd3f0b44b2c6e7aa34ccaa26targetblanka5cdd4ea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/tranmanhha135/Treg_uNK/tree/a5cdd4ec7e940249cd3f0b44b2c6e7aa34ccaa26" target="_blank">a5cdd4e</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  .DS_Store
    Untracked:  .gitignore
    Untracked:  0_data/.DS_Store
    Untracked:  0_data/rawData/.DS_Store
    Untracked:  0_data/rawData/Jiang_TableS2_41421_2022_513_MOESM3_ESM.csv
    Untracked:  0_data/rawData/MGI_tissues.tsv
    Untracked:  0_data/rawData/Table S1 13578_2022_749_MOESM1_ESM (1).xlsx
    Untracked:  0_data/rds_objects/cellchat.qs
    Untracked:  0_data/rds_objects/cellchat_veh.qs
    Untracked:  1_analysis/cellChat.Rmd
    Untracked:  2_plots/.DS_Store
    Untracked:  2_plots/LRheat.svg
    Untracked:  2_plots/mgi_hmap_1.svg
    Untracked:  2_plots/test.svg
    Untracked:  2_plots/trophoblast_hmap_1.png
    Untracked:  2_plots/trophoblast_hmap_1.svg
    Untracked:  2_plots/trophoblast_hmap_2.svg
    Untracked:  2_plots/trophoblast_hmap_3.svg
    Untracked:  2_plots/trophoblast_hmap_4.svg
    Untracked:  2_plots/trophoblast_hmap_5.svg
    Untracked:  3_output/sig_de_JiangSC_subset.xlsx
    Untracked:  sampleHeatmap.rds

Unstaged changes:
    Modified:   0_data/rds_objects/comp.rds
    Modified:   0_data/rds_objects/dge.rds
    Modified:   0_data/rds_objects/disease_function_sig.rds
    Modified:   0_data/rds_objects/ipa_pathways.rds
    Modified:   0_data/rds_objects/lm.rds
    Modified:   0_data/rds_objects/lm_all.rds
    Modified:   0_data/rds_objects/lm_sig.rds
    Modified:   0_data/rds_objects/upstream_sig.rds
    Modified:   0_data/rds_plots/deHmap_plots.rds
    Modified:   0_data/rds_plots/go_combined_dotPlot.rds
    Modified:   0_data/rds_plots/go_combined_parTerm_dotPlot.rds
    Modified:   0_data/rds_plots/go_dotPlot.rds
    Modified:   0_data/rds_plots/go_parTerm_dotPlot.rds
    Modified:   0_data/rds_plots/go_parTerm_scatter.rds
    Modified:   0_data/rds_plots/kegg_combined_dotPlot.rds
    Modified:   0_data/rds_plots/kegg_dotPlot.rds
    Modified:   0_data/rds_plots/kegg_path_Hmap.rds
    Modified:   0_data/rds_plots/ma_plots.rds
    Modified:   0_data/rds_plots/react_combined_dotPlot.rds
    Modified:   0_data/rds_plots/react_dotPlot.rds
    Modified:   0_data/rds_plots/vol_plots.rds
    Modified:   140_treg_uNK.Rproj
    Modified:   1_analysis/_site.yml
    Modified:   1_analysis/deAnalysis.Rmd
    Modified:   1_analysis/extraFigs.Rmd
    Modified:   1_analysis/go.Rmd
    Modified:   1_analysis/ipa.Rmd
    Modified:   1_analysis/kegg.Rmd
    Modified:   1_analysis/reactome.Rmd
    Modified:   1_analysis/setUp.Rmd
    Modified:   2_plots/1_QC/PC1_PC2.svg
    Modified:   2_plots/1_QC/PC1_PC3.svg
    Modified:   2_plots/1_QC/PC2_PC3.svg
    Modified:   2_plots/1_QC/counts_after_filtering.svg
    Modified:   2_plots/1_QC/counts_before_after_filtering.svg
    Modified:   2_plots/1_QC/counts_before_filtering.svg
    Modified:   2_plots/1_QC/library_size.svg
    Modified:   2_plots/2_DE/heat_DT vs veh.svg
    Modified:   2_plots/2_DE/heat_DT+Treg vs DT.svg
    Modified:   2_plots/2_DE/heat_DT+Treg vs veh.svg
    Modified:   2_plots/2_DE/ma_DT vs veh.png
    Modified:   2_plots/2_DE/ma_DT+Treg vs DT.png
    Modified:   2_plots/2_DE/ma_DT+Treg vs veh.png
    Modified:   2_plots/2_DE/venn.png
    Modified:   2_plots/2_DE/venn_2comp.png
    Modified:   2_plots/2_DE/vol_DT vs veh.png
    Modified:   2_plots/2_DE/vol_DT+Treg vs DT.png
    Modified:   2_plots/2_DE/vol_DT+Treg vs veh.png
    Modified:   2_plots/3_FA/go/combine_go_dot.svg
    Modified:   2_plots/3_FA/go/dot_DT vs veh.svg
    Modified:   2_plots/3_FA/go/dot_DT+Treg vs DT.svg
    Modified:   2_plots/3_FA/go/dot_DT+Treg vs veh.svg
    Modified:   2_plots/3_FA/go/parTerm_dot_DT vs veh.svg
    Modified:   2_plots/3_FA/go/parTerm_dot_DT+Treg vs DT.svg
    Modified:   2_plots/3_FA/go/parTerm_dot_DT+Treg vs veh.svg
    Modified:   2_plots/3_FA/go/parentTerm_all.png
    Modified:   2_plots/3_FA/go/semSim_dendrogram_DT vs veh.svg
    Modified:   2_plots/3_FA/go/semSim_dendrogram_DT+Treg vs DT.svg
    Modified:   2_plots/3_FA/go/semSim_dendrogram_DT+Treg vs veh.svg
    Modified:   2_plots/3_FA/go/semSim_scatter_DT vs veh.svg
    Modified:   2_plots/3_FA/go/semSim_scatter_DT+Treg vs DT.svg
    Modified:   2_plots/3_FA/go/semSim_scatter_DT+Treg vs veh.svg
    Modified:   2_plots/3_FA/go/upset_DT vs veh.svg
    Modified:   2_plots/3_FA/go/upset_DT+Treg vs DT.svg
    Modified:   2_plots/3_FA/go/upset_DT+Treg vs veh.svg
    Modified:   2_plots/3_FA/ipa/combine_ipa_dot.png
    Modified:   2_plots/3_FA/ipa/dnf_dot.png
    Modified:   2_plots/3_FA/ipa/upstream_hmap.png
    Modified:   2_plots/3_FA/ipa/venn_ipaDiseFunct.png
    Modified:   2_plots/3_FA/kegg/combine_kegg_dot.svg
    Modified:   2_plots/3_FA/kegg/heat_Antigen processing and presentation.svg
    Modified:   2_plots/3_FA/kegg/heat_Epstein-Barr virus infection.svg
    Modified:   2_plots/3_FA/kegg/heat_Natural killer cell mediated cytotoxicity.svg
    Modified:   2_plots/3_FA/kegg/heat_Phagosome.svg
    Modified:   2_plots/3_FA/kegg/kegg_dot_DT vs veh.svg
    Modified:   2_plots/3_FA/kegg/kegg_dot_DT+Treg vs DT.svg
    Modified:   2_plots/3_FA/kegg/kegg_dot_DT+Treg vs veh.svg
    Modified:   2_plots/3_FA/kegg/kegg_upset_DT vs veh.svg
    Modified:   2_plots/3_FA/kegg/kegg_upset_DT+Treg vs DT.svg
    Modified:   2_plots/3_FA/kegg/kegg_upset_DT+Treg vs veh.svg
    Modified:   2_plots/3_FA/reactome/combine_react_dot.svg
    Modified:   2_plots/3_FA/reactome/react_dot_DT vs veh.svg
    Modified:   2_plots/3_FA/reactome/react_dot_DT+Treg vs DT.svg
    Modified:   2_plots/3_FA/reactome/react_dot_DT+Treg vs veh.svg
    Modified:   2_plots/3_FA/reactome/react_upset_DT vs veh.svg
    Modified:   2_plots/3_FA/reactome/react_upset_DT+Treg vs DT.svg
    Modified:   2_plots/3_FA/reactome/react_upset_DT+Treg vs veh.svg
    Modified:   2_plots/3_FA/reactome/venn_react.png
    Modified:   2_plots/functionalHeat.png
    Modified:   2_plots/sampleHeat.svg
    Modified:   3_output/GO_sig.xlsx
    Modified:   3_output/KEGG_all.xlsx
    Modified:   3_output/KEGG_sig.xlsx
    Modified:   3_output/de_genes_all.xlsx
    Modified:   3_output/de_genes_sig.xlsx
    Modified:   3_output/reactome_all.xlsx
    Modified:   3_output/reactome_sig.xlsx
    Modified:   functions.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>1_analysis/extraFigs.Rmd</code>) and HTML
(<code>docs/extraFigs.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tranmanhha135/Treg_uNK/blob/a5cdd4ec7e940249cd3f0b44b2c6e7aa34ccaa26/1_analysis/extraFigs.Rmd" target="_blank">a5cdd4e</a>
</td>
<td>
git
</td>
<td>
2024-03-25
</td>
<td>
switching os
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tranmanhha135/Treg_uNK/blob/d0ea132b940b015bb5268b385fca13688cbe43b4/1_analysis/extraFigs.Rmd" target="_blank">d0ea132</a>
</td>
<td>
Ha Manh Tran
</td>
<td>
2024-02-15
</td>
<td>
windows crashed
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tranmanhha135/Treg_uNK/e9e7671a3bb7cdda8602bf9275a15e119c622cf1/docs/extraFigs.html" target="_blank">e9e7671</a>
</td>
<td>
tranmanhha135
</td>
<td>
2024-02-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tranmanhha135/Treg_uNK/blob/8da2e3114f4a95c7b48ce13a193ecaf617398137/1_analysis/extraFigs.Rmd" target="_blank">8da2e31</a>
</td>
<td>
tranmanhha135
</td>
<td>
2024-02-08
</td>
<td>
workflowr::wflow_publish(here::here("1_analysis/*.Rmd"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tranmanhha135/Treg_uNK/d8d23ee6420fd4e8d10d13b172c9812f6fcb1949/docs/extraFigs.html" target="_blank">d8d23ee</a>
</td>
<td>
tranmanhha135
</td>
<td>
2024-01-13
</td>
<td>
im on holiday
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tranmanhha135/Treg_uNK/36aeb85cdb811728c883b48c30fbbc564943ca3c/docs/extraFigs.html" target="_blank">36aeb85</a>
</td>
<td>
Ha Manh Tran
</td>
<td>
2024-01-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tranmanhha135/Treg_uNK/blob/a957cffecc77bb00bb12bd564a9acf69ac763907/1_analysis/extraFigs.Rmd" target="_blank">a957cff</a>
</td>
<td>
Ha Manh Tran
</td>
<td>
2024-01-13
</td>
<td>
workflowr::wflow_publish(here::here("1_analysis/*Rmd"))
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tranmanhha135/Treg_uNK/blob/c78dfac2aad574f5112c4ea94ad08fa6d760c9ca/1_analysis/extraFigs.Rmd" target="_blank">c78dfac</a>
</td>
<td>
tranmanhha135
</td>
<td>
2024-01-12
</td>
<td>
remote from ipad
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code># working with data
library(dplyr)
library(magrittr)
library(readr)
library(tibble)
library(reshape2)
library(tidyverse)
library(readxl)
library(showtext)
library(qs)

# Visualisation:
library(kableExtra)
library(ggplot2)
library(grid)
library(DT)
library(extrafont)
library(VennDiagram)

# Custom ggplot
library(ggplotify)
library(ComplexHeatmap)
library(gridExtra)
library(ggbiplot)
library(ggrepel)
library(rrvgo)
library(plotly)
library(GOSemSim)
library(data.table)
library(igraph)

# Bioconductor packages:
library(RColorBrewer)
library(CellChat)
library(edgeR)
library(limma)
library(Glimma)
library(clusterProfiler)
library(org.Mm.eg.db)
library(enrichplot)
library(patchwork)

library(pandoc)
library(knitr)
opts_knit$set(progress = FALSE, verbose = FALSE)
opts_chunk$set(warning=FALSE, message=FALSE, echo=FALSE)

# test &lt;- ggplot(cars,aes(speed,dist))+ geom_point()
# 
# ggsave(filename = &quot;test.svg&quot;, plot = test, path = here::here(&quot;2_plots/&quot;))
# ggsave(filename = &quot;sampleHeat.svg&quot;,plot = sampleHeatmap,path = here::here(&quot;2_plots/&quot;),width = 25.5,height = 20,units = &quot;cm&quot;)</code></pre>
<div id="functional-heatmap" class="section level1">
<h1>Functional heatmap</h1>
<p><img src="assets/functional_hmap.png" style="display: block; margin: auto;" /></p>
</div>
<div id="kegg-reactome" class="section level1">
<h1>KEGG + Reactome</h1>
<p><img src="assets/kegg_react.png" style="display: block; margin: auto;" /></p>
</div>
<div id="cellchatdb-jia-peng-et-al-2022-sc-paper"
class="section level1 tabset tabset-pills">
<h1 class="tabset tabset-pills">CellchatDB &amp; Jia-Peng et al 2022 sc
paper</h1>
<p>See <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8817544/">Jia-Peng et
al (2022)</a> paper for more details</p>
<div id="cellchatdb-subsetted-by-jia-peng-et-al-2022"
class="section level2">
<h2>CellchatDB subsetted by Jia-Peng et al 2022</h2>
<div class="datatables html-widget html-fill-item" id="htmlwidget-32cfd9db97395c4a5fd7" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-32cfd9db97395c4a5fd7">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"caption":"<caption>Table 1: Potential T-NK cell-cell communication from Jia-Peng et al (2022)<\/caption>","data":[["1","2","3","4","5","6"],["H2-t23  - (Klrd1+Klrc1)","H2-t23  - (Klrd1+Klrc2)","Thy1  - (Itgam+Itgb2)","Thy1  - (Itgax+Itgb2)","Thy1  - (Itgax+Itgb2)","Thy1 - Adgre5"],["MHC-I","MHC-I","THY1","THY1","THY1","ADGRE"],["H2-T23","H2-T23","Thy1","Thy1","Thy1","Thy1"],["Klrd1","Klrd1","Itgb2","Itgax","Itgb2","Adgre5"],["PMID: 24223577","PMID: 24223577","PMID: 15850796","PMID: 15850796","PMID: 15850796","PMID: 31462748"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>interaction_name_2<\/th>\n      <th>pathway_name<\/th>\n      <th>ligand.symbol_2<\/th>\n      <th>receptor.symbol_2<\/th>\n      <th>evidence<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","scrollX":true,"buttons":["copy","csv","excel","pdf","print"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"interaction_name_2","targets":1},{"name":"pathway_name","targets":2},{"name":"ligand.symbol_2","targets":3},{"name":"receptor.symbol_2","targets":4},{"name":"evidence","targets":5}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
<div class="datatables html-widget html-fill-item" id="htmlwidget-01554a25665d09d09f27" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-01554a25665d09d09f27">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"caption":"<caption>Table 2: Potential T-pNK cell-cell communication from Jia-Peng et al (2022)<\/caption>","data":[["1","2","3","4"],["H2-t23  - (Klrd1+Klrc1)","H2-t23  - (Klrd1+Klrc2)","Thy1  - (Itgam+Itgb2)","Thy1  - (Itgax+Itgb2)"],["MHC-I","MHC-I","THY1","THY1"],["H2-T23","H2-T23","Thy1","Thy1"],["Klrd1","Klrd1","Itgb2","Itgb2"],["PMID: 24223577","PMID: 24223577","PMID: 15850796","PMID: 15850796"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>interaction_name_2<\/th>\n      <th>pathway_name<\/th>\n      <th>ligand.symbol_2<\/th>\n      <th>receptor.symbol_2<\/th>\n      <th>evidence<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","scrollX":true,"buttons":["copy","csv","excel","pdf","print"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"interaction_name_2","targets":1},{"name":"pathway_name","targets":2},{"name":"ligand.symbol_2","targets":3},{"name":"receptor.symbol_2","targets":4},{"name":"evidence","targets":5}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
<div class="datatables html-widget html-fill-item" id="htmlwidget-a6251d00cbd347f90b02" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-a6251d00cbd347f90b02">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"caption":"<caption>Table 3: Potential NK-T cell-cell communication from Jia-Peng et al (2022)<\/caption>","data":[["1","2","3","4","5","6","7","8","9"],["Ifng  - (Ifngr1+Ifngr2)","Spp1  - Cd44","Spp1  - (Itga4+Itgb1)","Lamb3  - Cd44","Cdh1  - (Itgae+Itgb7)","Thy1  - (Itgam+Itgb2)","Thy1  - (Itgax+Itgb2)","Thy1 - Adgre5","(Itgav+Itgb1) - Adgre5"],["IFN-II","SPP1","SPP1","LAMININ","CDH1","THY1","THY1","ADGRE","ADGRE"],["Ifng","Spp1","Spp1","Lamb3","Cdh1","Thy1","Thy1","Thy1","Itgb1"],["Ifngr1","Cd44","Itga4","Cd44","Itgb7","Itgb2","Itgb2","Adgre5","Adgre5"],["KEGG: mmu04060","PMID: 21907263","PMID: 21907263","KEGG: mmu04512","PMID: 9425167","PMID: 15850796","PMID: 15850796","PMID: 31462748","PMID: 31462748"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>interaction_name_2<\/th>\n      <th>pathway_name<\/th>\n      <th>ligand.symbol_2<\/th>\n      <th>receptor.symbol_2<\/th>\n      <th>evidence<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","scrollX":true,"buttons":["copy","csv","excel","pdf","print"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"interaction_name_2","targets":1},{"name":"pathway_name","targets":2},{"name":"ligand.symbol_2","targets":3},{"name":"receptor.symbol_2","targets":4},{"name":"evidence","targets":5}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="heatmaps" class="section level2">
<h2>Heatmaps</h2>
<p><img src="figure/extraFigs.Rmd/cellChatDB_heat-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="figure/extraFigs.Rmd/cellChatDB_heat_logFC-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="jiang-et-al-2023-trophoblast-sc-paper"
class="section level1 tabset tabset-pills">
<h1 class="tabset tabset-pills">Jiang et al (2023) trophoblast SC
paper</h1>
<p>See <a
href="https://www.nature.com/articles/s41421-022-00513-z">Jiang et al
(2023)</a> for full paper</p>
<div id="large-heatmaps" class="section level2 tabset tabset-pills">
<h2 class="tabset tabset-pills">Large Heatmaps</h2>
<p>Large heatmaps contains all significant DEGs that matched genes
associated with known trophoblast cell types identified in Jiang et al
2023 from <code>DT vs veh</code> <strong>or</strong>
<code>DT+Treg vs DT</code>. Therefore DEGs that are COMMON and UNIQUE to
both comparisons.</p>
<div id="heatmap-1" class="section level3">
<h3>Heatmap 1</h3>
<p>LogCPM of all 127 significant DEGs that are expressed by various
trophoblasts cell types identified in Jiang et al (2023) paper. Cell
type annotations not include as that will be too overwhelming.</p>
<p><img src="figure/extraFigs.Rmd/hmap1-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="heatmap-2" class="section level3">
<h3>Heatmap 2</h3>
<p>Grey means that gene was not significantly differentially expressed
in that comparison.</p>
<p><img src="assets/trophoblast_large_hmap2.png" style="display: block; margin: auto;" /></p>
</div>
<div id="heatmap-3" class="section level3">
<h3>Heatmap 3</h3>
<p>Heatmap with all logFC values where significance indicated by
asterisks</p>
<p><img src="assets/trophoblast_large_hmap3.png" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="small-heatmaps" class="section level2 tabset tabset-pills">
<h2 class="tabset tabset-pills">Small Heatmaps</h2>
<p>These heatmaps contains only DEGs that were significant in BOTH DT vs
veh AND DT+Treg vs DT.</p>
<div id="heatmap-1-1" class="section level3">
<h3>Heatmap 1</h3>
<p><img src="assets/trophoblast_small_hmap1.png" style="display: block; margin: auto;" /></p>
</div>
<div id="heatmap-2-1" class="section level3">
<h3>Heatmap 2</h3>
<p><img src="assets/trophoblast_small_hmap2.png" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="mgi-data-combined" class="section level1">
<h1>MGI data combined</h1>
<p>This includes significant DEGs that matched genes associated with
known expression in placenta BUT NOT in uterus (based on MGI gene
expression data)</p>
<p>Of these 38 genes, those which were also found in the sc trophoblast
paper from Jiang et al 2023 paper were also annotated</p>
<p>IMPORTANTLY, this version contains only DEGs that were significant in
BOTH DT vs veh AND DT+Treg vs DT.</p>
<p>Again, note that logFC were included for all DEGs but the significant
ones in each comparison is marked by asterisk.</p>
<p>ONE CAVEAT: Since this list of 38 genes were obtained
programmatically and not manually, this list may contains genes which
may have only a few publications which supports it. I suggest using
these 38 and going back to MGI for more exploration.</p>
<p><img src="assets/mgi_hmap.png" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.4.1 (2024-06-14)
Platform: aarch64-apple-darwin20
Running under: macOS Sonoma 14.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Australia/Adelaide
tzcode source: internal

attached base packages:
[1] stats4    grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] knitr_1.48             pandoc_0.2.0           patchwork_1.2.0       
 [4] enrichplot_1.24.2      org.Mm.eg.db_3.19.1    AnnotationDbi_1.66.0  
 [7] IRanges_2.38.1         S4Vectors_0.42.1       clusterProfiler_4.12.2
[10] Glimma_2.14.0          edgeR_4.2.1            limma_3.60.4          
[13] CellChat_2.1.2         Biobase_2.64.0         BiocGenerics_0.50.0   
[16] RColorBrewer_1.1-3     igraph_2.0.3           data.table_1.15.4     
[19] GOSemSim_2.30.0        plotly_4.10.4          rrvgo_1.16.0          
[22] ggrepel_0.9.5.9999     ggbiplot_0.6.2         gridExtra_2.3         
[25] ComplexHeatmap_2.20.0  ggplotify_0.1.2        VennDiagram_1.7.3     
[28] futile.logger_1.4.3    extrafont_0.19         DT_0.33               
[31] kableExtra_1.4.0       qs_0.26.3              showtext_0.9-7        
[34] showtextdb_3.0         sysfonts_0.8.9         readxl_1.4.3          
[37] lubridate_1.9.3        forcats_1.0.0          stringr_1.5.1         
[40] purrr_1.0.2            tidyr_1.3.1            ggplot2_3.5.1         
[43] tidyverse_2.0.0        reshape2_1.4.4         tibble_3.2.1          
[46] readr_2.1.5            magrittr_2.0.3         dplyr_1.1.4           

loaded via a namespace (and not attached):
  [1] vroom_1.6.5                 Biostrings_2.72.1          
  [3] vctrs_0.6.5                 RApiSerialize_0.1.3        
  [5] digest_0.6.36               png_0.1-8                  
  [7] shape_1.4.6.1               registry_0.5-1             
  [9] git2r_0.33.0                parallelly_1.38.0          
 [11] magick_2.8.4                MASS_7.3-61                
 [13] httpuv_1.6.15               foreach_1.5.2              
 [15] qvalue_2.36.0               withr_3.0.1                
 [17] xfun_0.46                   ggfun_0.1.5                
 [19] ggpubr_0.6.0                memoise_2.0.1              
 [21] gson_0.1.0                  systemfonts_1.1.0          
 [23] ragg_1.3.2                  tidytree_0.4.6             
 [25] GlobalOptions_0.1.2         pbapply_1.7-2              
 [27] KEGGREST_1.44.1             promises_1.3.0             
 [29] httr_1.4.7                  rstatix_0.7.2              
 [31] globals_0.16.3              stringfish_0.16.0          
 [33] rstudioapi_0.16.0           UCSC.utils_1.0.0           
 [35] generics_0.1.3              DOSE_3.30.2                
 [37] ggalluvial_0.12.5           zlibbioc_1.50.0            
 [39] ggraph_2.2.1                polyclip_1.10-7            
 [41] GenomeInfoDbData_1.2.12     SparseArray_1.4.8          
 [43] xtable_1.8-4                doParallel_1.0.17          
 [45] evaluate_0.24.0             S4Arrays_1.4.1             
 [47] hms_1.1.3                   GenomicRanges_1.56.1       
 [49] irlba_2.3.5.1               colorspace_2.1-1           
 [51] ggnetwork_0.5.13            NLP_0.2-1                  
 [53] reticulate_1.38.0           treemap_2.4-4              
 [55] later_1.3.2                 viridis_0.6.5              
 [57] ggtree_3.12.0               lattice_0.22-6             
 [59] NMF_0.27                    future.apply_1.11.2        
 [61] shadowtext_0.1.4            cowplot_1.1.3              
 [63] matrixStats_1.3.0           pillar_1.9.0               
 [65] nlme_3.1-165                iterators_1.0.14           
 [67] sna_2.7-2                   gridBase_0.4-7             
 [69] compiler_4.4.1              RSpectra_0.16-2            
 [71] stringi_1.8.4               SummarizedExperiment_1.34.0
 [73] plyr_1.8.9                  crayon_1.5.3               
 [75] abind_1.4-5                 gridGraphics_0.5-1         
 [77] locfit_1.5-9.10             graphlayouts_1.1.1         
 [79] bit_4.0.5                   fastmatch_1.1-4            
 [81] whisker_0.4.1               codetools_0.2-20           
 [83] textshaping_0.4.0           openssl_2.2.0              
 [85] crosstalk_1.2.1             bslib_0.8.0                
 [87] slam_0.1-52                 GetoptLong_1.0.5           
 [89] tm_0.7-13                   mime_0.12                  
 [91] splines_4.4.1               circlize_0.4.16            
 [93] Rcpp_1.0.13                 HDO.db_0.99.1              
 [95] cellranger_1.1.0            Rttf2pt1_1.3.12            
 [97] blob_1.2.4                  utf8_1.2.4                 
 [99] here_1.0.1                  clue_0.3-65                
[101] fs_1.6.4                    listenv_0.9.1              
[103] ggsignif_0.6.4              Matrix_1.7-0               
[105] statmod_1.5.0               tzdb_0.4.0                 
[107] svglite_2.1.3               tweenr_2.0.3               
[109] pkgconfig_2.0.3             pheatmap_1.0.12            
[111] network_1.18.2              tools_4.4.1                
[113] cachem_1.1.0                RSQLite_2.3.7              
[115] viridisLite_0.4.2           DBI_1.2.3                  
[117] fastmap_1.2.0               rmarkdown_2.27             
[119] scales_1.3.0                broom_1.0.6                
[121] sass_0.4.9                  coda_0.19-4.1              
[123] FNN_1.1.4                   carData_3.0-5              
[125] farver_2.1.2                tidygraph_1.3.1            
[127] scatterpie_0.2.3            yaml_2.3.10                
[129] workflowr_1.7.1             MatrixGenerics_1.16.0      
[131] cli_3.6.3                   lifecycle_1.0.4            
[133] askpass_1.2.0               lambda.r_1.2.4             
[135] backports_1.5.0             BiocParallel_1.38.0        
[137] timechange_0.3.0            gtable_0.3.5               
[139] rjson_0.2.21                umap_0.2.10.0              
[141] parallel_4.4.1              ape_5.8                    
[143] jsonlite_1.8.8              bit64_4.0.5                
[145] yulab.utils_0.1.5           BiocNeighbors_1.22.0       
[147] RcppParallel_5.1.8          futile.options_1.0.1       
[149] jquerylib_0.1.4             highr_0.11                 
[151] lazyeval_0.2.2              shiny_1.9.1                
[153] htmltools_0.5.8.1           GO.db_3.19.1               
[155] rappdirs_0.3.3              formatR_1.14               
[157] glue_1.7.0                  XVector_0.44.0             
[159] rprojroot_2.0.4             treeio_1.28.0              
[161] extrafontdb_1.0             R6_2.5.1                   
[163] DESeq2_1.44.0               labeling_0.4.3             
[165] cluster_2.1.6               rngtools_1.5.2             
[167] wordcloud_2.6               aplot_0.2.3                
[169] GenomeInfoDb_1.40.1         statnet.common_4.9.0       
[171] DelayedArray_0.30.1         tidyselect_1.2.1           
[173] ggforce_0.4.2               xml2_1.3.6                 
[175] car_3.1-2                   future_1.34.0              
[177] munsell_0.5.1               htmlwidgets_1.6.4          
[179] fgsea_1.30.0                rlang_1.1.4                
[181] fansi_1.0.6                </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
