<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Ha Tran" />

<meta name="date" content="2024-01-12" />

<title>Extra figures</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Treg-uNK</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="setUp.html">Pre-processing</a>
</li>
<li>
  <a href="deAnalysis.html">DE Analysis</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Over-representation Analysis (ORA)
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="go.html">Gene Ontology</a>
    </li>
    <li>
      <a href="kegg.html">KEGG</a>
    </li>
    <li>
      <a href="reactome.html">Reactome</a>
    </li>
    <li>
      <a href="ipa.html">IPA</a>
    </li>
  </ul>
</li>
<li>
  <a href="extraFigs.html">Extra figures</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://www.linkedin.com/in/ha-tran-6973b5130/">
    <span class="fa fa-linkedin"></span>
     
  </a>
</li>
<li>
  <a href="https://github.com/tranmanhha135/">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Extra figures</h1>
<h4 class="author">Ha Tran</h4>
<h4 class="date">2024-01-12</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-02-08
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Treg_uNK/1_analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed12345code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(12345)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed12345code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(12345)</code> was run prior to running the
code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtranmanhha135TreguNKtree8da2e3114f4a95c7b48ce13a193ecaf617398137targetblank8da2e31a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/tranmanhha135/Treg_uNK/tree/8da2e3114f4a95c7b48ce13a193ecaf617398137" target="_blank">8da2e31</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomtranmanhha135TreguNKtree8da2e3114f4a95c7b48ce13a193ecaf617398137targetblank8da2e31a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/tranmanhha135/Treg_uNK/tree/8da2e3114f4a95c7b48ce13a193ecaf617398137" target="_blank">8da2e31</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    2_plots/4_GSEA/

Untracked files:
    Untracked:  .gitignore
    Untracked:  0_data/functions/compColour.rds
    Untracked:  0_data/rawData/DTvsPBS_upstream regulators.xls
    Untracked:  0_data/rawData/IPA Diseases and Functions.xlsx
    Untracked:  0_data/rawData/IPA pathways.xlsx
    Untracked:  0_data/rawData/IPA upstream regulators.xlsx
    Untracked:  0_data/rawData/Treg vs DT_diseases and functions.xls
    Untracked:  0_data/rawData/Treg vs DT_pathways.xls
    Untracked:  0_data/rawData/Treg vs DT_upstream regulators.xls
    Untracked:  0_data/rawData/Treg vs PBS_diseases and functions.xls
    Untracked:  0_data/rawData/Treg vs PBS_pathways.xls
    Untracked:  0_data/rawData/Treg vs PBS_upstream regulators.xls
    Untracked:  0_data/rds_objects/disease_function_sig.rds
    Untracked:  0_data/rds_objects/ipa_pathways.rds
    Untracked:  0_data/rds_objects/upstream_sig.rds
    Untracked:  0_data/rds_plots/
    Untracked:  2_plots/2_DE/heat_DT vs veh.svg
    Untracked:  2_plots/2_DE/heat_DT+Treg vs DT.svg
    Untracked:  2_plots/2_DE/heat_DT+Treg vs veh.svg
    Untracked:  2_plots/2_DE/hist_DT vs veh.svg
    Untracked:  2_plots/2_DE/hist_DT+Treg vs DT.svg
    Untracked:  2_plots/2_DE/hist_DT+Treg vs veh.svg
    Untracked:  2_plots/2_DE/ma_DT vs veh.png
    Untracked:  2_plots/2_DE/ma_DT+Treg vs DT.png
    Untracked:  2_plots/2_DE/ma_DT+Treg vs veh.png
    Untracked:  2_plots/2_DE/venn_2comp.png
    Untracked:  2_plots/2_DE/venn_2comp.png.2024-02-02_16-30-41.893776.log
    Untracked:  2_plots/2_DE/venn_2comp.png.2024-02-02_16-31-08.009748.log
    Untracked:  2_plots/2_DE/vol_DT vs veh.png
    Untracked:  2_plots/2_DE/vol_DT+Treg vs DT.png
    Untracked:  2_plots/2_DE/vol_DT+Treg vs veh.png
    Untracked:  2_plots/3_FA/go/dot_DT vs veh.svg
    Untracked:  2_plots/3_FA/go/dot_DT+Treg vs DT.svg
    Untracked:  2_plots/3_FA/go/dot_DT+Treg vs veh.svg
    Untracked:  2_plots/3_FA/go/parTerm_dot_DT vs veh.svg
    Untracked:  2_plots/3_FA/go/parTerm_dot_DT+Treg vs DT.svg
    Untracked:  2_plots/3_FA/go/parTerm_dot_DT+Treg vs veh.svg
    Untracked:  2_plots/3_FA/go/semSim_dendrogram_DT vs veh.svg
    Untracked:  2_plots/3_FA/go/semSim_dendrogram_DT+Treg vs DT.svg
    Untracked:  2_plots/3_FA/go/semSim_dendrogram_DT+Treg vs veh.svg
    Untracked:  2_plots/3_FA/go/semSim_scatter_DT vs veh.svg
    Untracked:  2_plots/3_FA/go/semSim_scatter_DT+Treg vs DT.svg
    Untracked:  2_plots/3_FA/go/semSim_scatter_DT+Treg vs veh.svg
    Untracked:  2_plots/3_FA/go/upset_DT vs veh.svg
    Untracked:  2_plots/3_FA/go/upset_DT+Treg vs DT.svg
    Untracked:  2_plots/3_FA/go/upset_DT+Treg vs veh.svg
    Untracked:  2_plots/3_FA/ipa/
    Untracked:  2_plots/3_FA/kegg/heat_Antigen processing and presentation.svg
    Untracked:  2_plots/3_FA/kegg/heat_Epstein-Barr virus infection.svg
    Untracked:  2_plots/3_FA/kegg/heat_Natural killer cell mediated cytotoxicity.svg
    Untracked:  2_plots/3_FA/kegg/heat_Phagosome.svg
    Untracked:  2_plots/3_FA/kegg/kegg_dot_DT vs veh.svg
    Untracked:  2_plots/3_FA/kegg/kegg_dot_DT+Treg vs DT.svg
    Untracked:  2_plots/3_FA/kegg/kegg_dot_DT+Treg vs veh.svg
    Untracked:  2_plots/3_FA/kegg/kegg_upset_DT vs veh.svg
    Untracked:  2_plots/3_FA/kegg/kegg_upset_DT+Treg vs DT.svg
    Untracked:  2_plots/3_FA/kegg/kegg_upset_DT+Treg vs veh.svg
    Untracked:  2_plots/3_FA/kegg/mmu04145.png
    Untracked:  2_plots/3_FA/kegg/mmu04145.xml
    Untracked:  2_plots/3_FA/kegg/mmu04612.png
    Untracked:  2_plots/3_FA/kegg/mmu04612.xml
    Untracked:  2_plots/3_FA/kegg/mmu04650.png
    Untracked:  2_plots/3_FA/kegg/mmu04650.xml
    Untracked:  2_plots/3_FA/kegg/mmu05169.png
    Untracked:  2_plots/3_FA/kegg/mmu05169.xml
    Untracked:  2_plots/3_FA/kegg/pv_mmu04145.png
    Untracked:  2_plots/3_FA/kegg/pv_mmu04612.png
    Untracked:  2_plots/3_FA/kegg/pv_mmu04650.png
    Untracked:  2_plots/3_FA/kegg/pv_mmu05169.png
    Untracked:  2_plots/3_FA/reactome/combine_react_dot.svg
    Untracked:  2_plots/3_FA/reactome/react_dot_DT vs veh.svg
    Untracked:  2_plots/3_FA/reactome/react_dot_DT+Treg vs DT.svg
    Untracked:  2_plots/3_FA/reactome/react_dot_DT+Treg vs veh.svg
    Untracked:  2_plots/3_FA/reactome/react_upset_DT vs veh.svg
    Untracked:  2_plots/3_FA/reactome/react_upset_DT+Treg vs DT.svg
    Untracked:  2_plots/3_FA/reactome/react_upset_DT+Treg vs veh.svg
    Untracked:  2_plots/3_FA/reactome/venn_react.png

Unstaged changes:
    Modified:   0_data/functions/bossTheme.rds
    Modified:   0_data/functions/bossTheme_bar.rds
    Modified:   0_data/functions/groupColour.rds
    Modified:   0_data/functions/groupColour_dark.rds
    Modified:   0_data/rds_objects/comp.rds
    Modified:   0_data/rds_objects/dge.rds
    Modified:   0_data/rds_objects/enrichGO.rds
    Modified:   0_data/rds_objects/enrichGO_sig.rds
    Modified:   0_data/rds_objects/enrichKEGG.rds
    Modified:   0_data/rds_objects/enrichKEGG_all.rds
    Modified:   0_data/rds_objects/enrichKEGG_sig.rds
    Modified:   0_data/rds_objects/lm.rds
    Modified:   0_data/rds_objects/lm_all.rds
    Modified:   0_data/rds_objects/lm_sig.rds
    Modified:   0_data/rds_objects/pathway_details.rds
    Modified:   0_data/rds_objects/rawCount.rds
    Modified:   0_data/rds_objects/reactome.rds
    Modified:   0_data/rds_objects/reactome_all.rds
    Modified:   0_data/rds_objects/reactome_sig.rds
    Modified:   0_data/rds_objects/reducedTerms_all.rds
    Modified:   0_data/rds_objects/reducedTerms_ora.rds
    Modified:   0_data/rds_objects/reduced_semSim_df.rds
    Modified:   0_data/rds_objects/scores_ora.rds
    Modified:   0_data/rds_objects/semSim_df.rds
    Modified:   0_data/rds_objects/simMatrix_ora.rds
    Modified:   140_treg_uNK.Rproj
    Modified:   1_analysis/_site.yml
    Modified:   2_plots/1_QC/PC1_PC2.svg
    Modified:   2_plots/1_QC/PC1_PC3.svg
    Modified:   2_plots/1_QC/PC2_PC3.svg
    Modified:   2_plots/1_QC/counts_after_filtering.svg
    Modified:   2_plots/1_QC/counts_before_after_filtering.svg
    Modified:   2_plots/1_QC/counts_before_filtering.svg
    Modified:   2_plots/1_QC/library_size.svg
    Deleted:    2_plots/2_DE/heat_DT vs PBS.svg
    Deleted:    2_plots/2_DE/heat_Treg vs DT.svg
    Deleted:    2_plots/2_DE/heat_Treg vs PBS.svg
    Modified:   2_plots/2_DE/heat_combined.svg
    Deleted:    2_plots/2_DE/hist_DT vs PBS.svg
    Deleted:    2_plots/2_DE/hist_Treg vs DT.svg
    Deleted:    2_plots/2_DE/hist_Treg vs PBS.svg
    Deleted:    2_plots/2_DE/ma_DT vs PBS.png
    Deleted:    2_plots/2_DE/ma_Treg vs DT.png
    Deleted:    2_plots/2_DE/ma_Treg vs PBS.png
    Modified:   2_plots/2_DE/venn.png
    Deleted:    2_plots/2_DE/venn.png.2024-01-08_19-59-50.400111.log
    Deleted:    2_plots/2_DE/vol_DT vs PBS.png
    Deleted:    2_plots/2_DE/vol_Treg vs DT.png
    Deleted:    2_plots/2_DE/vol_Treg vs PBS.png
    Modified:   2_plots/3_FA/go/combine_go_dot.svg
    Deleted:    2_plots/3_FA/go/dot_DT vs PBS.svg
    Deleted:    2_plots/3_FA/go/dot_Treg vs DT.svg
    Deleted:    2_plots/3_FA/go/dot_Treg vs PBS.svg
    Modified:   2_plots/3_FA/go/parTerm_BP_venn.png
    Modified:   2_plots/3_FA/go/parTerm_CC_venn.png
    Modified:   2_plots/3_FA/go/parTerm_MF_venn.png
    Deleted:    2_plots/3_FA/go/parTerm_dot_DT vs PBS.svg
    Deleted:    2_plots/3_FA/go/parTerm_dot_Treg vs DT.svg
    Deleted:    2_plots/3_FA/go/parTerm_dot_Treg vs PBS.svg
    Modified:   2_plots/3_FA/go/parentTerm_all.svg
    Deleted:    2_plots/3_FA/go/upset_DT vs PBS.svg
    Deleted:    2_plots/3_FA/go/upset_Treg vs DT.svg
    Deleted:    2_plots/3_FA/go/upset_Treg vs PBS.svg
    Modified:   2_plots/3_FA/go/venn.png
    Modified:   2_plots/3_FA/kegg/combine_kegg_dot.svg
    Deleted:    2_plots/3_FA/kegg/heat_DT vs PBS_Antigen processing and presentation.svg
    Deleted:    2_plots/3_FA/kegg/heat_DT vs PBS_Natural killer cell mediated cytotoxicity.svg
    Deleted:    2_plots/3_FA/kegg/heat_DT vs PBS_Phagosome.svg
    Deleted:    2_plots/3_FA/kegg/heat_DT vs PBS_Th1 and Th2 cell differentiation.svg
    Deleted:    2_plots/3_FA/kegg/heat_Treg vs DT_Antigen processing and presentation.svg
    Deleted:    2_plots/3_FA/kegg/heat_Treg vs DT_Natural killer cell mediated cytotoxicity.svg
    Deleted:    2_plots/3_FA/kegg/heat_Treg vs DT_Phagosome.svg
    Deleted:    2_plots/3_FA/kegg/heat_Treg vs DT_Th1 and Th2 cell differentiation.svg
    Deleted:    2_plots/3_FA/kegg/heat_Treg vs PBS_Antigen processing and presentation.svg
    Deleted:    2_plots/3_FA/kegg/heat_Treg vs PBS_Natural killer cell mediated cytotoxicity.svg
    Deleted:    2_plots/3_FA/kegg/heat_Treg vs PBS_Phagosome.svg
    Deleted:    2_plots/3_FA/kegg/heat_Treg vs PBS_Th1 and Th2 cell differentiation.svg
    Deleted:    2_plots/3_FA/kegg/kegg_dot_DT vs PBS.svg
    Deleted:    2_plots/3_FA/kegg/kegg_dot_Treg vs DT.svg
    Deleted:    2_plots/3_FA/kegg/kegg_dot_Treg vs PBS.svg
    Deleted:    2_plots/3_FA/kegg/kegg_upset_DT vs PBS.svg
    Deleted:    2_plots/3_FA/kegg/kegg_upset_Treg vs DT.svg
    Deleted:    2_plots/3_FA/kegg/kegg_upset_Treg vs PBS.svg
    Modified:   2_plots/3_FA/kegg/venn.png
    Deleted:    2_plots/3_FA/reactome/react_dot_DT vs PBS.svg
    Deleted:    2_plots/3_FA/reactome/react_dot_Treg vs DT.svg
    Deleted:    2_plots/3_FA/reactome/react_dot_Treg vs PBS.svg
    Deleted:    2_plots/3_FA/reactome/react_upset_DT vs PBS.svg
    Deleted:    2_plots/3_FA/reactome/react_upset_Treg vs DT.svg
    Deleted:    2_plots/3_FA/reactome/react_upset_Treg vs PBS.svg
    Modified:   2_plots/functionalHeat.svg
    Modified:   3_output/GO_sig.xlsx
    Modified:   3_output/KEGG_all.xlsx
    Modified:   3_output/KEGG_sig.xlsx
    Modified:   3_output/de_genes_all.xlsx
    Modified:   3_output/de_genes_sig.xlsx
    Modified:   3_output/reactome_all.xlsx
    Modified:   3_output/reactome_sig.xlsx
    Modified:   functions.Rmd
    Deleted:    ipa.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>1_analysis/extraFigs.Rmd</code>) and HTML
(<code>docs/extraFigs.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tranmanhha135/Treg_uNK/blob/8da2e3114f4a95c7b48ce13a193ecaf617398137/1_analysis/extraFigs.Rmd" target="_blank">8da2e31</a>
</td>
<td>
tranmanhha135
</td>
<td>
2024-02-08
</td>
<td>
workflowr::wflow_publish(here::here("1_analysis/*.Rmd"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tranmanhha135/Treg_uNK/d8d23ee6420fd4e8d10d13b172c9812f6fcb1949/docs/extraFigs.html" target="_blank">d8d23ee</a>
</td>
<td>
tranmanhha135
</td>
<td>
2024-01-13
</td>
<td>
im on holiday
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tranmanhha135/Treg_uNK/36aeb85cdb811728c883b48c30fbbc564943ca3c/docs/extraFigs.html" target="_blank">36aeb85</a>
</td>
<td>
Ha Manh Tran
</td>
<td>
2024-01-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tranmanhha135/Treg_uNK/blob/a957cffecc77bb00bb12bd564a9acf69ac763907/1_analysis/extraFigs.Rmd" target="_blank">a957cff</a>
</td>
<td>
Ha Manh Tran
</td>
<td>
2024-01-13
</td>
<td>
workflowr::wflow_publish(here::here("1_analysis/*Rmd"))
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tranmanhha135/Treg_uNK/blob/c78dfac2aad574f5112c4ea94ad08fa6d760c9ca/1_analysis/extraFigs.Rmd" target="_blank">c78dfac</a>
</td>
<td>
tranmanhha135
</td>
<td>
2024-01-12
</td>
<td>
remote from ipad
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code># working with data
library(dplyr)
library(magrittr)
library(readr)
library(tibble)
library(reshape2)
library(tidyverse)

# Visualisation:
library(kableExtra)
library(ggplot2)
library(grid)
library(DT)
library(extrafont)
library(VennDiagram)

# Custom ggplot
library(gridExtra)
library(ggbiplot)
library(ggrepel)
library(rrvgo)
library(d3treeR)
library(plotly)
library(GOSemSim)
library(data.table)

# Bioconductor packages:
library(edgeR)
library(limma)
library(Glimma)
library(clusterProfiler)
library(org.Mm.eg.db)
library(enrichplot)
library(patchwork)

library(pandoc)
library(knitr)
opts_knit$set(progress = FALSE, verbose = FALSE)
opts_chunk$set(warning=FALSE, message=FALSE, echo=FALSE)</code></pre>
<div id="extra-figures" class="section level2">
<h2>Extra Figures</h2>
<pre><code> [1] &quot;Granzyme-mediated programmed cell death\nsignaling pathway&quot;
 [2] &quot;T cell mediated cytotoxicity&quot;                              
 [3] &quot;Cytolysis&quot;                                                 
 [4] &quot;Natural killer cell mediated cytotoxicity&quot;                 
 [5] &quot;Regulation of leukocyte mediated cytotoxicity&quot;             
 [6] &quot;Natural killer cell lectin-like receptor\nbinding&quot;         
 [7] &quot;Antigen processing and presentation&quot;                       
 [8] &quot;ER-Phagosome pathway&quot;                                      
 [9] &quot;Interferon gamma signaling&quot;                                
[10] &quot;Interferon alpha/beta signaling&quot;                           
[11] &quot;Neutrophil degranulation&quot;                                  </code></pre>
<pre><code>[1] &quot;Gzmb&quot;  &quot;Gzmc&quot;  &quot;Gzmd&quot;  &quot;Gzme&quot;  &quot;Gzmf&quot;  &quot;Gzmg&quot;  &quot;Lamp1&quot; &quot;Prf1&quot;  &quot;Srgn&quot; 
 [1] &quot;B2m&quot;      &quot;Ctsc&quot;     &quot;Gzmb&quot;     &quot;H2-D1&quot;    &quot;H2-K1&quot;    &quot;H2-Q4&quot;   
 [7] &quot;H2-T23&quot;   &quot;Prf1&quot;     &quot;Ptprc&quot;    &quot;Serpinb9&quot; &quot;Tap2&quot;    
[1] &quot;Gzmb&quot; &quot;Gzmc&quot; &quot;Gzmd&quot; &quot;Gzme&quot; &quot;Gzmf&quot; &quot;Gzmg&quot; &quot;Prf1&quot;
 [1] &quot;Gzmb&quot;     &quot;Gzmc&quot;     &quot;Gzmd&quot;     &quot;Gzme&quot;     &quot;Gzmf&quot;     &quot;Gzmg&quot;    
 [7] &quot;Havcr2&quot;   &quot;Lamp1&quot;    &quot;Serpinb9&quot; &quot;Tap1&quot;     &quot;Tap2&quot;    
 [1] &quot;B2m&quot;      &quot;H2-D1&quot;    &quot;H2-K1&quot;    &quot;H2-Q4&quot;    &quot;H2-T23&quot;   &quot;Havcr2&quot;  
 [7] &quot;Lamp1&quot;    &quot;Ptprc&quot;    &quot;Serpinb9&quot; &quot;Tap1&quot;     &quot;Tap2&quot;    
[1] &quot;H2-D1&quot;  &quot;H2-K1&quot;  &quot;H2-Q4&quot;  &quot;H2-T23&quot;
 [1] &quot;B2m&quot;    &quot;H2-D1&quot;  &quot;H2-K1&quot;  &quot;H2-Q4&quot;  &quot;H2-T23&quot; &quot;Hspa1b&quot; &quot;Psme1&quot;  &quot;Psme2&quot; 
 [9] &quot;Tap1&quot;   &quot;Tap2&quot;  
[1] &quot;B2m&quot;    &quot;H2-K1&quot;  &quot;H2-Q4&quot;  &quot;H2-T23&quot; &quot;Tap1&quot;   &quot;Tap2&quot;  
 [1] &quot;B2m&quot;    &quot;Camk2d&quot; &quot;Gbp2&quot;   &quot;Gbp4&quot;   &quot;Gbp6&quot;   &quot;HLA-A&quot;  &quot;HLA-E&quot;  &quot;Icam1&quot; 
 [9] &quot;Irf7&quot;   &quot;Irf8&quot;   &quot;OAS1&quot;   &quot;Oas2&quot;   &quot;Stat1&quot;  &quot;Vcam1&quot; 
 [1] &quot;Gbp2&quot;  &quot;HLA-A&quot; &quot;HLA-E&quot; &quot;Irf7&quot;  &quot;Irf8&quot;  &quot;OAS1&quot;  &quot;Oas2&quot;  &quot;Psmb8&quot; &quot;Stat1&quot;
[10] &quot;Stat2&quot;
 [1] &quot;Adam8&quot;    &quot;Arhgap45&quot; &quot;B2m&quot;      &quot;Bin2&quot;     &quot;Cd55&quot;     &quot;Cd63&quot;    
 [7] &quot;Ctsc&quot;     &quot;Ctsg&quot;     &quot;Dock2&quot;    &quot;Dpp7&quot;     &quot;Lamp1&quot;    &quot;Olr1&quot;    
[13] &quot;Ptprc&quot;    &quot;Pygl&quot;     &quot;SERPINA1&quot; &quot;Tbc1d10c&quot;</code></pre>
<pre><code>[1] &quot;Gzmc&quot;  &quot;Gzmd&quot;  &quot;Gzme&quot;  &quot;Gzmf&quot;  &quot;Gzmg&quot;  &quot;Lamp1&quot;
[1] &quot;Gzmc&quot; &quot;Gzmd&quot; &quot;Gzme&quot; &quot;Gzmf&quot; &quot;Gzmg&quot;
[1] &quot;Gzmc&quot;  &quot;Gzmd&quot;  &quot;Gzme&quot;  &quot;Gzmf&quot;  &quot;Gzmg&quot;  &quot;Lamp1&quot;
[1] &quot;B2m&quot;    &quot;H2-D1&quot;  &quot;H2-K1&quot;  &quot;H2-T23&quot;
[1] &quot;B2m&quot;    &quot;H2-D1&quot;  &quot;H2-K1&quot;  &quot;H2-T23&quot; &quot;Lamp1&quot; 
[1] &quot;H2-D1&quot;  &quot;H2-K1&quot;  &quot;H2-T23&quot;
[1] &quot;B2m&quot;    &quot;H2-D1&quot;  &quot;H2-K1&quot;  &quot;H2-T23&quot; &quot;Hspa1l&quot; &quot;Rfxap&quot; 
[1] &quot;B2m&quot;    &quot;H2-K1&quot;  &quot;H2-T23&quot;
[1] &quot;B2m&quot;   &quot;Gbp4&quot;  &quot;Gbp6&quot;  &quot;HLA-A&quot; &quot;HLA-E&quot; &quot;Irf8&quot; 
[1] &quot;HLA-A&quot; &quot;HLA-E&quot; &quot;Irf8&quot;  &quot;Stat2&quot;
[1] &quot;Adam8&quot; &quot;Aldoc&quot; &quot;B2m&quot;   &quot;Bin2&quot;  &quot;FTL&quot;   &quot;Lamp1&quot;</code></pre>
<pre><code>png 
  2 </code></pre>
<p><img src="figure/extraFigs.Rmd/functionalHeat-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-functionalHeat-1">
Past versions of functionalHeat-1.png
</button>
</p>
<div id="fig-functionalHeat-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tranmanhha135/Treg_uNK/blob/36aeb85cdb811728c883b48c30fbbc564943ca3c/docs/figure/extraFigs.Rmd/functionalHeat-1.png" target="_blank">36aeb85</a>
</td>
<td>
Ha Manh Tran
</td>
<td>
2024-01-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.3.2 (2023-10-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.3 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0 
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

time zone: Australia/Adelaide
tzcode source: system (glibc)

attached base packages:
[1] stats4    grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ggplotify_0.1.2       RColorBrewer_1.1-3    ComplexHeatmap_2.16.0
 [4] knitr_1.45            pandoc_0.2.0          patchwork_1.1.3      
 [7] enrichplot_1.20.3     org.Mm.eg.db_3.17.0   AnnotationDbi_1.62.2 
[10] IRanges_2.34.1        S4Vectors_0.38.2      Biobase_2.60.0       
[13] BiocGenerics_0.46.0   clusterProfiler_4.8.3 Glimma_2.10.0        
[16] edgeR_3.42.4          limma_3.56.2          data.table_1.14.10   
[19] GOSemSim_2.26.1       plotly_4.10.3         d3treeR_0.1          
[22] rrvgo_1.12.2          ggrepel_0.9.4         ggbiplot_0.55        
[25] scales_1.3.0          plyr_1.8.9            gridExtra_2.3        
[28] VennDiagram_1.7.3     futile.logger_1.4.3   extrafont_0.19       
[31] DT_0.31               kableExtra_1.3.4      lubridate_1.9.3      
[34] forcats_1.0.0         stringr_1.5.1         purrr_1.0.2          
[37] tidyr_1.3.0           ggplot2_3.4.4         tidyverse_2.0.0      
[40] reshape2_1.4.4        tibble_3.2.1          readr_2.1.4          
[43] magrittr_2.0.3        dplyr_1.1.4          

loaded via a namespace (and not attached):
  [1] fs_1.6.3                    matrixStats_1.2.0          
  [3] bitops_1.0-7                HDO.db_0.99.1              
  [5] httr_1.4.7                  webshot_0.5.5              
  [7] doParallel_1.0.17           tools_4.3.2                
  [9] utf8_1.2.4                  R6_2.5.1                   
 [11] lazyeval_0.2.2              GetoptLong_1.0.5           
 [13] withr_2.5.2                 cli_3.6.2                  
 [15] formatR_1.14                scatterpie_0.2.1           
 [17] labeling_0.4.3              slam_0.1-50                
 [19] sass_0.4.8                  tm_0.7-11                  
 [21] askpass_1.2.0               systemfonts_1.0.5          
 [23] yulab.utils_0.1.2           gson_0.1.0                 
 [25] DOSE_3.26.2                 svglite_2.1.3              
 [27] rstudioapi_0.15.0           RSQLite_2.3.4              
 [29] treemap_2.4-4               generics_0.1.3             
 [31] gridGraphics_0.5-1          shape_1.4.6                
 [33] GO.db_3.17.0                Matrix_1.6-3               
 [35] fansi_1.0.6                 abind_1.4-5                
 [37] lifecycle_1.0.4             whisker_0.4.1              
 [39] yaml_2.3.8                  SummarizedExperiment_1.30.2
 [41] qvalue_2.32.0               blob_1.2.4                 
 [43] promises_1.2.1              crayon_1.5.2               
 [45] lattice_0.22-5              cowplot_1.1.2              
 [47] KEGGREST_1.40.1             magick_2.8.2               
 [49] pillar_1.9.0                fgsea_1.26.0               
 [51] GenomicRanges_1.52.1        rjson_0.2.21               
 [53] codetools_0.2-19            fastmatch_1.1-4            
 [55] glue_1.6.2                  downloader_0.4             
 [57] ggfun_0.1.3                 vctrs_0.6.5                
 [59] png_0.1-8                   treeio_1.24.3              
 [61] gtable_0.3.4                cachem_1.0.8               
 [63] xfun_0.41                   S4Arrays_1.0.6             
 [65] mime_0.12                   tidygraph_1.3.0            
 [67] pheatmap_1.0.12             iterators_1.0.14           
 [69] ellipsis_0.3.2              nlme_3.1-163               
 [71] ggtree_3.8.2                bit64_4.0.5                
 [73] GenomeInfoDb_1.36.4         rprojroot_2.0.4            
 [75] data.tree_1.1.0             bslib_0.6.1                
 [77] colorspace_2.1-0            DBI_1.2.0                  
 [79] DESeq2_1.40.2               tidyselect_1.2.0           
 [81] bit_4.0.5                   compiler_4.3.2             
 [83] extrafontdb_1.0             git2r_0.33.0               
 [85] rvest_1.0.3                 xml2_1.3.6                 
 [87] NLP_0.2-1                   DelayedArray_0.26.7        
 [89] shadowtext_0.1.2            rappdirs_0.3.3             
 [91] digest_0.6.33               rmarkdown_2.25             
 [93] XVector_0.40.0              htmltools_0.5.7            
 [95] pkgconfig_2.0.3             umap_0.2.10.0              
 [97] MatrixGenerics_1.12.3       highr_0.10                 
 [99] fastmap_1.1.1               rlang_1.1.2                
[101] GlobalOptions_0.1.2         htmlwidgets_1.6.4          
[103] shiny_1.8.0                 farver_2.1.1               
[105] jquerylib_0.1.4             jsonlite_1.8.8             
[107] BiocParallel_1.34.2         RCurl_1.98-1.13            
[109] GenomeInfoDbData_1.2.10     wordcloud_2.6              
[111] munsell_0.5.0               Rcpp_1.0.11                
[113] ape_5.7-1                   viridis_0.6.4              
[115] reticulate_1.34.0           stringi_1.8.3              
[117] ggraph_2.1.0                zlibbioc_1.46.0            
[119] MASS_7.3-60                 parallel_4.3.2             
[121] Biostrings_2.68.1           graphlayouts_1.0.2         
[123] splines_4.3.2               hms_1.1.3                  
[125] circlize_0.4.15             locfit_1.5-9.8             
[127] igraph_1.6.0                futile.options_1.0.1       
[129] XML_3.99-0.16               evaluate_0.23              
[131] lambda.r_1.2.4              tzdb_0.4.0                 
[133] foreach_1.5.2               tweenr_2.0.2               
[135] httpuv_1.6.13               Rttf2pt1_1.3.12            
[137] openssl_2.1.1               polyclip_1.10-6            
[139] clue_0.3-65                 gridBase_0.4-7             
[141] ggforce_0.4.1               xtable_1.8-4               
[143] RSpectra_0.16-1             tidytree_0.4.6             
[145] later_1.3.2                 viridisLite_0.4.2          
[147] aplot_0.2.2                 memoise_2.0.1              
[149] cluster_2.1.6               workflowr_1.7.1            
[151] timechange_0.2.0            here_1.0.1                 
[153] gridSVG_1.7-5              </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
